<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Page not found | Introductory Bioinformatics for VAI Users</title>
  <meta name="description" content="Bioinformatics tutorials with hands-on exercises and examples." />
  <meta name="generator" content="bookdown 0.42 and GitBook 2.6.7" />

  <meta property="og:title" content="Page not found | Introductory Bioinformatics for VAI Users" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Bioinformatics tutorials with hands-on exercises and examples." />
  <meta name="github-repo" content="vari-bbc/Intro_to_Linux_for_HPC" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Page not found | Introductory Bioinformatics for VAI Users" />
  
  <meta name="twitter:description" content="Bioinformatics tutorials with hands-on exercises and examples." />
  

<meta name="author" content="The Bioinformatics and Biostatistics Core (BBC)" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  


<script src="assets/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="assets/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="assets/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="assets/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="assets/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="assets/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="assets/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="assets/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="assets/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="assets/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="assets/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="assets/kePrint-0.0.1/kePrint.js"></script>
<link href="assets/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> <strong>Introduction</strong></a></li>
<li class="chapter" data-level="2" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html"><i class="fa fa-check"></i><b>2</b> <strong>Basic Command Line Utilities on the HPC</strong></a>
<ul>
<li class="chapter" data-level="2.1" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#access-hpc"><i class="fa fa-check"></i><b>2.1</b> <strong>Access HPC</strong></a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#ondemand-instructions"><i class="fa fa-check"></i><b>2.1.1</b> <strong>OnDemand Instructions</strong></a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#file-navigation"><i class="fa fa-check"></i><b>2.2</b> <strong>File navigation</strong></a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#navigate-to-a-directory-folder"><i class="fa fa-check"></i><b>2.2.1</b> <strong>Navigate to a directory (folder)</strong></a></li>
<li class="chapter" data-level="2.2.2" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#list-content-in-a-directory"><i class="fa fa-check"></i><b>2.2.2</b> <strong>List content in a directory</strong></a></li>
<li class="chapter" data-level="2.2.3" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#navigate-back-to-home-directory"><i class="fa fa-check"></i><b>2.2.3</b> <strong>Navigate back to home directory</strong></a></li>
<li class="chapter" data-level="2.2.4" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#create-a-directory"><i class="fa fa-check"></i><b>2.2.4</b> <strong>Create a directory</strong></a></li>
<li class="chapter" data-level="2.2.5" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#copy-a-file"><i class="fa fa-check"></i><b>2.2.5</b> <strong>Copy a file</strong></a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#view-and-manipulate-files"><i class="fa fa-check"></i><b>2.3</b> <strong>View and manipulate files</strong></a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#display-content"><i class="fa fa-check"></i><b>2.3.1</b> <strong>Display content</strong></a></li>
<li class="chapter" data-level="2.3.2" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#pattern-search"><i class="fa fa-check"></i><b>2.3.2</b> <strong>Pattern search</strong></a></li>
<li class="chapter" data-level="2.3.3" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#display-the-number-of-lines-words-and-characters"><i class="fa fa-check"></i><b>2.3.3</b> <strong>Display the number of lines, words, and characters</strong></a></li>
<li class="chapter" data-level="2.3.4" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#pipe---redirection"><i class="fa fa-check"></i><b>2.3.4</b> <strong>Pipe - redirection</strong></a></li>
<li class="chapter" data-level="2.3.5" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#output-redirection"><i class="fa fa-check"></i><b>2.3.5</b> <strong>Output redirection</strong></a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#exercise"><i class="fa fa-check"></i><b>2.4</b> <strong>Exercise</strong></a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#step-1-copy-the-files"><i class="fa fa-check"></i><b>2.4.1</b> <strong>Step 1: Copy the files</strong></a></li>
<li class="chapter" data-level="2.4.2" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#step-2-combine-the-files"><i class="fa fa-check"></i><b>2.4.2</b> <strong>Step 2: Combine the files</strong></a></li>
<li class="chapter" data-level="2.4.3" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#step-3-count-the-number-of-reads"><i class="fa fa-check"></i><b>2.4.3</b> <strong>Step 3: Count the number of reads</strong></a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#summary"><i class="fa fa-check"></i><b>2.5</b> <strong>Summary</strong></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="mini-project.html"><a href="mini-project.html"><i class="fa fa-check"></i><b>3</b> <strong>Mini Project</strong></a>
<ul>
<li class="chapter" data-level="3.1" data-path="mini-project.html"><a href="mini-project.html#start-an-interactive-job"><i class="fa fa-check"></i><b>3.1</b> <strong>Start an Interactive Job</strong></a></li>
<li class="chapter" data-level="3.2" data-path="mini-project.html"><a href="mini-project.html#create-directory"><i class="fa fa-check"></i><b>3.2</b> <strong>Create Directory</strong></a></li>
<li class="chapter" data-level="3.3" data-path="mini-project.html"><a href="mini-project.html#copy-fastq-files-to-fastqs-subdirectory"><i class="fa fa-check"></i><b>3.3</b> <strong>Copy Fastq Files To <code>fastqs</code> subdirectory</strong></a></li>
<li class="chapter" data-level="3.4" data-path="mini-project.html"><a href="mini-project.html#use-basic-linux-commands-to-explore-the-fastq-files"><i class="fa fa-check"></i><b>3.4</b> <strong>Use basic Linux commands to explore the fastq files</strong></a></li>
<li class="chapter" data-level="3.5" data-path="mini-project.html"><a href="mini-project.html#working-with-environment-modules"><i class="fa fa-check"></i><b>3.5</b> <strong>Working With Environment Modules</strong></a></li>
<li class="chapter" data-level="3.6" data-path="mini-project.html"><a href="mini-project.html#run-fastqc"><i class="fa fa-check"></i><b>3.6</b> <strong>Run FastQC</strong></a></li>
<li class="chapter" data-level="3.7" data-path="mini-project.html"><a href="mini-project.html#set-up-a-job-script-to-run-salmon-to-align-the-reads"><i class="fa fa-check"></i><b>3.7</b> <strong>Set up a job script to run Salmon to align the reads</strong></a></li>
<li class="chapter" data-level="3.8" data-path="mini-project.html"><a href="mini-project.html#submit-a-job"><i class="fa fa-check"></i><b>3.8</b> <strong>Submit a Job</strong></a></li>
<li class="chapter" data-level="3.9" data-path="mini-project.html"><a href="mini-project.html#examine-the-standard-output-stdout-and-standard-error-stderr-log-files"><i class="fa fa-check"></i><b>3.9</b> <strong>Examine the Standard Output (stdout) and Standard Error (stderr) log files</strong></a></li>
<li class="chapter" data-level="3.10" data-path="mini-project.html"><a href="mini-project.html#use-grep-to-find-the-tpms-for-a-specific-gene"><i class="fa fa-check"></i><b>3.10</b> <strong>Use grep to find the TPMs for a specific gene</strong></a></li>
<li class="chapter" data-level="3.11" data-path="mini-project.html"><a href="mini-project.html#bonus-use-an-interactive-job-to-run-multiqc-on-the-salmon-and-fastqc-output"><i class="fa fa-check"></i><b>3.11</b> <strong>BONUS: Use an interactive job to run multiQC on the Salmon and FastQC output</strong></a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><i class="fa fa-check"></i><b>4</b> <strong>Basic RNA-Seq Analysis – From fastq files to DE genes</strong></a>
<ul>
<li class="chapter" data-level="4.1" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#run-the-bbc-pipeline-to-align-the-reads"><i class="fa fa-check"></i><b>4.1</b> <strong>Run the BBC pipeline to align the reads</strong></a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#the-workflow-as-a-graph"><i class="fa fa-check"></i><b>4.1.1</b> <strong>The workflow as a graph</strong></a></li>
<li class="chapter" data-level="4.1.2" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#clone-the-github-repo"><i class="fa fa-check"></i><b>4.1.2</b> <strong>Clone the Github repo</strong></a></li>
<li class="chapter" data-level="4.1.3" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#add-your-fastq-files-to-raw_data"><i class="fa fa-check"></i><b>4.1.3</b> <strong>Add your fastq files to <code>raw_data/</code></strong></a></li>
<li class="chapter" data-level="4.1.4" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#fill-out-the-samplesheet"><i class="fa fa-check"></i><b>4.1.4</b> <strong>Fill out the samplesheet</strong></a></li>
<li class="chapter" data-level="4.1.5" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#fill-out-the-config-file"><i class="fa fa-check"></i><b>4.1.5</b> <strong>Fill out the config file</strong></a></li>
<li class="chapter" data-level="4.1.6" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#submit-the-main-snakemake-job"><i class="fa fa-check"></i><b>4.1.6</b> <strong>Submit the main Snakemake job</strong></a></li>
<li class="chapter" data-level="4.1.7" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#bbc-maintained-reference-files"><i class="fa fa-check"></i><b>4.1.7</b> <strong>BBC-maintained reference files</strong></a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#symlink-pre-run-results-to-your-directory"><i class="fa fa-check"></i><b>4.2</b> <strong>Symlink pre-run results to your directory</strong></a></li>
<li class="chapter" data-level="4.3" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#quality-control"><i class="fa fa-check"></i><b>4.3</b> <strong>Quality control</strong></a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#multiqc-report"><i class="fa fa-check"></i><b>4.3.1</b> <strong>multiQC report</strong></a></li>
<li class="chapter" data-level="4.3.2" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#other-considerations"><i class="fa fa-check"></i><b>4.3.2</b> <strong>Other considerations</strong></a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#de-analysis-using-r-and-deseq2"><i class="fa fa-check"></i><b>4.4</b> <strong>DE analysis using R and DESeq2</strong></a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#create-an-r-project"><i class="fa fa-check"></i><b>4.4.1</b> <strong>Create an R project</strong></a></li>
<li class="chapter" data-level="4.4.2" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#exploring-the-data-using-isee"><i class="fa fa-check"></i><b>4.4.2</b> <strong>Exploring the data using iSEE</strong></a></li>
<li class="chapter" data-level="4.4.3" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#make-a-separate-folder-for-de-analysis-and-a-script-file-to-store-our-commands"><i class="fa fa-check"></i><b>4.4.3</b> <strong>Make a separate folder for DE analysis and a script file to store our commands</strong></a></li>
<li class="chapter" data-level="4.4.4" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#make-an-output-directory"><i class="fa fa-check"></i><b>4.4.4</b> <strong>Make an output directory</strong></a></li>
<li class="chapter" data-level="4.4.5" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#install-packages"><i class="fa fa-check"></i><b>4.4.5</b> <strong>Install packages</strong></a></li>
<li class="chapter" data-level="4.4.6" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#load-packages"><i class="fa fa-check"></i><b>4.4.6</b> <strong>Load packages</strong></a></li>
<li class="chapter" data-level="4.4.7" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#set-up-your-deseq-object"><i class="fa fa-check"></i><b>4.4.7</b> <strong>Set up your DESeq object</strong></a></li>
<li class="chapter" data-level="4.4.8" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#what-is-a-summarizedexperiment-object"><i class="fa fa-check"></i><b>4.4.8</b> <strong>What is a SummarizedExperiment object?</strong></a></li>
<li class="chapter" data-level="4.4.9" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#remove-genes-with-lowno-expression"><i class="fa fa-check"></i><b>4.4.9</b> <strong>Remove genes with low/no expression</strong></a></li>
<li class="chapter" data-level="4.4.10" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#different-normalization-approaches-for-different-biases"><i class="fa fa-check"></i><b>4.4.10</b> <strong>Different normalization approaches for different biases</strong></a></li>
<li class="chapter" data-level="4.4.11" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#run-the-de-workflow"><i class="fa fa-check"></i><b>4.4.11</b> <strong>Run the DE workflow</strong></a></li>
<li class="chapter" data-level="4.4.12" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#summarize-de-results"><i class="fa fa-check"></i><b>4.4.12</b> <strong>Summarize DE results</strong></a></li>
<li class="chapter" data-level="4.4.13" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#p-value-distribution"><i class="fa fa-check"></i><b>4.4.13</b> <strong>P value distribution</strong></a></li>
<li class="chapter" data-level="4.4.14" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#shrink-log-fold-changes-for-lowly-expressed-genes"><i class="fa fa-check"></i><b>4.4.14</b> <strong>Shrink log fold changes for lowly expressed genes</strong></a></li>
<li class="chapter" data-level="4.4.15" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#output-de-results"><i class="fa fa-check"></i><b>4.4.15</b> <strong>Output DE results</strong></a></li>
<li class="chapter" data-level="4.4.16" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#output-tables-with-log2-normalized-counts"><i class="fa fa-check"></i><b>4.4.16</b> <strong>Output tables with log2 normalized counts</strong></a></li>
<li class="chapter" data-level="4.4.17" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#why-should-we-transform-counts-for-visualizations-or-clustering"><i class="fa fa-check"></i><b>4.4.17</b> <strong>Why should we transform counts for visualizations or clustering?</strong></a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#volcano-plot"><i class="fa fa-check"></i><b>4.5</b> <strong>Volcano plot</strong></a></li>
<li class="chapter" data-level="4.6" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#make-a-new-summarizedexperiment-to-store-all-results"><i class="fa fa-check"></i><b>4.6</b> <strong>Make a new SummarizedExperiment to store all results</strong></a></li>
<li class="chapter" data-level="4.7" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#heatmap-of-top-de-genes"><i class="fa fa-check"></i><b>4.7</b> <strong>Heatmap of top DE genes</strong></a></li>
<li class="chapter" data-level="4.8" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#look-for-specific-genes"><i class="fa fa-check"></i><b>4.8</b> <strong>Look for specific genes</strong></a></li>
<li class="chapter" data-level="4.9" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#summarizedexperiment-exercises"><i class="fa fa-check"></i><b>4.9</b> <strong>SummarizedExperiment exercises</strong></a></li>
<li class="chapter" data-level="4.10" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#your-turn"><i class="fa fa-check"></i><b>4.10</b> <strong>Your turn</strong></a></li>
<li class="chapter" data-level="4.11" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#venn-diagrams"><i class="fa fa-check"></i><b>4.11</b> <strong>Venn diagrams</strong></a></li>
<li class="chapter" data-level="4.12" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#upset-plots"><i class="fa fa-check"></i><b>4.12</b> <strong>Upset plots</strong></a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="pathway-and-enrichment-analysis.html"><a href="pathway-and-enrichment-analysis.html"><i class="fa fa-check"></i><b>5</b> <strong>Pathway and enrichment analysis</strong></a>
<ul>
<li class="chapter" data-level="5.1" data-path="pathway-and-enrichment-analysis.html"><a href="pathway-and-enrichment-analysis.html#downstream-of-de-gene-analysis"><i class="fa fa-check"></i><b>5.1</b> <strong>Downstream of DE gene analysis</strong></a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="pathway-and-enrichment-analysis.html"><a href="pathway-and-enrichment-analysis.html#load-packages-1"><i class="fa fa-check"></i><b>5.1.1</b> <strong>load packages</strong></a></li>
<li class="chapter" data-level="5.1.2" data-path="pathway-and-enrichment-analysis.html"><a href="pathway-and-enrichment-analysis.html#import-previous-de-results"><i class="fa fa-check"></i><b>5.1.2</b> <strong>Import previous DE results</strong></a></li>
<li class="chapter" data-level="5.1.3" data-path="pathway-and-enrichment-analysis.html"><a href="pathway-and-enrichment-analysis.html#how-many-de-genes"><i class="fa fa-check"></i><b>5.1.3</b> <strong>How many DE genes</strong></a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="pathway-and-enrichment-analysis.html"><a href="pathway-and-enrichment-analysis.html#why-enrichment-and-pathway-analysis"><i class="fa fa-check"></i><b>5.2</b> <strong>Why enrichment and pathway analysis?</strong></a></li>
<li class="chapter" data-level="5.3" data-path="pathway-and-enrichment-analysis.html"><a href="pathway-and-enrichment-analysis.html#ora"><i class="fa fa-check"></i><b>5.3</b> <strong>ORA</strong></a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="pathway-and-enrichment-analysis.html"><a href="pathway-and-enrichment-analysis.html#plot-ora-results"><i class="fa fa-check"></i><b>5.3.1</b> <strong>plot ORA results</strong></a></li>
<li class="chapter" data-level="5.3.2" data-path="pathway-and-enrichment-analysis.html"><a href="pathway-and-enrichment-analysis.html#save-ora-results-to-a-tsv-file"><i class="fa fa-check"></i><b>5.3.2</b> <strong>save ORA results to a tsv file</strong></a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="pathway-and-enrichment-analysis.html"><a href="pathway-and-enrichment-analysis.html#gsea"><i class="fa fa-check"></i><b>5.4</b> <strong>GSEA</strong></a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="pathway-and-enrichment-analysis.html"><a href="pathway-and-enrichment-analysis.html#save-gsea-results"><i class="fa fa-check"></i><b>5.4.1</b> <strong>save gsea results</strong></a></li>
<li class="chapter" data-level="5.4.2" data-path="pathway-and-enrichment-analysis.html"><a href="pathway-and-enrichment-analysis.html#plot-gsea-results"><i class="fa fa-check"></i><b>5.4.2</b> <strong>plot GSEA results</strong></a></li>
<li class="chapter" data-level="5.4.3" data-path="pathway-and-enrichment-analysis.html"><a href="pathway-and-enrichment-analysis.html#plot-kegg-map"><i class="fa fa-check"></i><b>5.4.3</b> <strong>plot KEGG map</strong></a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="pathway-and-enrichment-analysis.html"><a href="pathway-and-enrichment-analysis.html#other-r-packages-and-tools"><i class="fa fa-check"></i><b>5.5</b> <strong>Other R packages and tools</strong></a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="appendices.html"><a href="appendices.html"><i class="fa fa-check"></i><b>6</b> <strong>Appendices</strong></a>
<ul>
<li class="chapter" data-level="6.1" data-path="appendices.html"><a href="appendices.html#other-workshop-files"><i class="fa fa-check"></i><b>6.1</b> <strong>Other workshop files</strong></a></li>
<li class="chapter" data-level="6.2" data-path="appendices.html"><a href="appendices.html#local-access-of-files-on-the-hpc"><i class="fa fa-check"></i><b>6.2</b> <strong>Local access of files on the HPC</strong></a></li>
<li class="chapter" data-level="6.3" data-path="appendices.html"><a href="appendices.html#bash-cheatsheet"><i class="fa fa-check"></i><b>6.3</b> <strong>Bash cheatsheet</strong></a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Introductory Bioinformatics for VAI Users</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="page-not-found" class="section level1">
<h1>Page not found</h1>
<p>The page you requested cannot be found (perhaps it was moved or renamed).</p>
<p>You may want to try searching to find the page's new location, or use
the table of contents to find the page you are looking for.</p>
</div>
            </section>

          </div>
        </div>
      </div>


    </div>
  </div>
<script src="assets/gitbook-2.6.7/js/app.min.js"></script>
<script src="assets/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="assets/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="assets/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="assets/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="assets/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="assets/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="assets/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": true,
    "facebook": false,
    "twitter": false,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": false
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": null,
    "text": null
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": null,
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  },
  "toolbar": {
    "position": "static"
  }
});
});
</script>

</body>

</html>
