<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Part 4 Basic RNA-Seq Analysis – From fastq files to DE genes | Introductory Bioinformatics for VAI Users</title>
  <meta name="description" content="Bioinformatics tutorials with hands-on exercises and examples." />
  <meta name="generator" content="bookdown 0.42 and GitBook 2.6.7" />

  <meta property="og:title" content="Part 4 Basic RNA-Seq Analysis – From fastq files to DE genes | Introductory Bioinformatics for VAI Users" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Bioinformatics tutorials with hands-on exercises and examples." />
  <meta name="github-repo" content="vari-bbc/Intro_to_Linux_for_HPC" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Part 4 Basic RNA-Seq Analysis – From fastq files to DE genes | Introductory Bioinformatics for VAI Users" />
  
  <meta name="twitter:description" content="Bioinformatics tutorials with hands-on exercises and examples." />
  

<meta name="author" content="The Bioinformatics and Biostatistics Core (BBC)" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="mini-project.html"/>
<link rel="next" href="pathway-and-enrichment-analysis.html"/>
<script src="assets/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="assets/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="assets/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="assets/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="assets/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="assets/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="assets/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="assets/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="assets/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="assets/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="assets/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="assets/kePrint-0.0.1/kePrint.js"></script>
<link href="assets/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> <strong>Introduction</strong></a></li>
<li class="chapter" data-level="2" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html"><i class="fa fa-check"></i><b>2</b> <strong>Basic Command Line Utilities on the HPC</strong></a>
<ul>
<li class="chapter" data-level="2.1" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#access-hpc"><i class="fa fa-check"></i><b>2.1</b> <strong>Access HPC</strong></a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#ondemand-instructions"><i class="fa fa-check"></i><b>2.1.1</b> <strong>OnDemand Instructions</strong></a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#file-navigation"><i class="fa fa-check"></i><b>2.2</b> <strong>File navigation</strong></a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#navigate-to-a-directory-folder"><i class="fa fa-check"></i><b>2.2.1</b> <strong>Navigate to a directory (folder)</strong></a></li>
<li class="chapter" data-level="2.2.2" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#list-content-in-a-directory"><i class="fa fa-check"></i><b>2.2.2</b> <strong>List content in a directory</strong></a></li>
<li class="chapter" data-level="2.2.3" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#navigate-back-to-home-directory"><i class="fa fa-check"></i><b>2.2.3</b> <strong>Navigate back to home directory</strong></a></li>
<li class="chapter" data-level="2.2.4" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#create-a-directory"><i class="fa fa-check"></i><b>2.2.4</b> <strong>Create a directory</strong></a></li>
<li class="chapter" data-level="2.2.5" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#copy-a-file"><i class="fa fa-check"></i><b>2.2.5</b> <strong>Copy a file</strong></a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#view-and-manipulate-files"><i class="fa fa-check"></i><b>2.3</b> <strong>View and manipulate files</strong></a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#display-content"><i class="fa fa-check"></i><b>2.3.1</b> <strong>Display content</strong></a></li>
<li class="chapter" data-level="2.3.2" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#pattern-search"><i class="fa fa-check"></i><b>2.3.2</b> <strong>Pattern search</strong></a></li>
<li class="chapter" data-level="2.3.3" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#display-the-number-of-lines-words-and-characters"><i class="fa fa-check"></i><b>2.3.3</b> <strong>Display the number of lines, words, and characters</strong></a></li>
<li class="chapter" data-level="2.3.4" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#pipe---redirection"><i class="fa fa-check"></i><b>2.3.4</b> <strong>Pipe - redirection</strong></a></li>
<li class="chapter" data-level="2.3.5" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#output-redirection"><i class="fa fa-check"></i><b>2.3.5</b> <strong>Output redirection</strong></a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#exercise"><i class="fa fa-check"></i><b>2.4</b> <strong>Exercise</strong></a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#step-1-copy-the-files"><i class="fa fa-check"></i><b>2.4.1</b> <strong>Step 1: Copy the files</strong></a></li>
<li class="chapter" data-level="2.4.2" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#step-2-combine-the-files"><i class="fa fa-check"></i><b>2.4.2</b> <strong>Step 2: Combine the files</strong></a></li>
<li class="chapter" data-level="2.4.3" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#step-3-count-the-number-of-reads"><i class="fa fa-check"></i><b>2.4.3</b> <strong>Step 3: Count the number of reads</strong></a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#summary"><i class="fa fa-check"></i><b>2.5</b> <strong>Summary</strong></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="mini-project.html"><a href="mini-project.html"><i class="fa fa-check"></i><b>3</b> <strong>Mini Project</strong></a>
<ul>
<li class="chapter" data-level="3.1" data-path="mini-project.html"><a href="mini-project.html#start-an-interactive-job"><i class="fa fa-check"></i><b>3.1</b> <strong>Start an Interactive Job</strong></a></li>
<li class="chapter" data-level="3.2" data-path="mini-project.html"><a href="mini-project.html#create-directory"><i class="fa fa-check"></i><b>3.2</b> <strong>Create Directory</strong></a></li>
<li class="chapter" data-level="3.3" data-path="mini-project.html"><a href="mini-project.html#copy-fastq-files-to-fastqs-subdirectory"><i class="fa fa-check"></i><b>3.3</b> <strong>Copy Fastq Files To <code>fastqs</code> subdirectory</strong></a></li>
<li class="chapter" data-level="3.4" data-path="mini-project.html"><a href="mini-project.html#use-basic-linux-commands-to-explore-the-fastq-files"><i class="fa fa-check"></i><b>3.4</b> <strong>Use basic Linux commands to explore the fastq files</strong></a></li>
<li class="chapter" data-level="3.5" data-path="mini-project.html"><a href="mini-project.html#working-with-environment-modules"><i class="fa fa-check"></i><b>3.5</b> <strong>Working With Environment Modules</strong></a></li>
<li class="chapter" data-level="3.6" data-path="mini-project.html"><a href="mini-project.html#run-fastqc"><i class="fa fa-check"></i><b>3.6</b> <strong>Run FastQC</strong></a></li>
<li class="chapter" data-level="3.7" data-path="mini-project.html"><a href="mini-project.html#set-up-a-job-script-to-run-salmon-to-align-the-reads"><i class="fa fa-check"></i><b>3.7</b> <strong>Set up a job script to run Salmon to align the reads</strong></a></li>
<li class="chapter" data-level="3.8" data-path="mini-project.html"><a href="mini-project.html#submit-a-job"><i class="fa fa-check"></i><b>3.8</b> <strong>Submit a Job</strong></a></li>
<li class="chapter" data-level="3.9" data-path="mini-project.html"><a href="mini-project.html#examine-the-standard-output-stdout-and-standard-error-stderr-log-files"><i class="fa fa-check"></i><b>3.9</b> <strong>Examine the Standard Output (stdout) and Standard Error (stderr) log files</strong></a></li>
<li class="chapter" data-level="3.10" data-path="mini-project.html"><a href="mini-project.html#use-grep-to-find-the-tpms-for-a-specific-gene"><i class="fa fa-check"></i><b>3.10</b> <strong>Use grep to find the TPMs for a specific gene</strong></a></li>
<li class="chapter" data-level="3.11" data-path="mini-project.html"><a href="mini-project.html#bonus-use-an-interactive-job-to-run-multiqc-on-the-salmon-and-fastqc-output"><i class="fa fa-check"></i><b>3.11</b> <strong>BONUS: Use an interactive job to run multiQC on the Salmon and FastQC output</strong></a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><i class="fa fa-check"></i><b>4</b> <strong>Basic RNA-Seq Analysis – From fastq files to DE genes</strong></a>
<ul>
<li class="chapter" data-level="4.1" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#run-the-bbc-pipeline-to-align-the-reads"><i class="fa fa-check"></i><b>4.1</b> <strong>Run the BBC pipeline to align the reads</strong></a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#the-workflow-as-a-graph"><i class="fa fa-check"></i><b>4.1.1</b> <strong>The workflow as a graph</strong></a></li>
<li class="chapter" data-level="4.1.2" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#clone-the-github-repo"><i class="fa fa-check"></i><b>4.1.2</b> <strong>Clone the Github repo</strong></a></li>
<li class="chapter" data-level="4.1.3" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#add-your-fastq-files-to-raw_data"><i class="fa fa-check"></i><b>4.1.3</b> <strong>Add your fastq files to <code>raw_data/</code></strong></a></li>
<li class="chapter" data-level="4.1.4" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#fill-out-the-samplesheet"><i class="fa fa-check"></i><b>4.1.4</b> <strong>Fill out the samplesheet</strong></a></li>
<li class="chapter" data-level="4.1.5" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#fill-out-the-config-file"><i class="fa fa-check"></i><b>4.1.5</b> <strong>Fill out the config file</strong></a></li>
<li class="chapter" data-level="4.1.6" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#submit-the-main-snakemake-job"><i class="fa fa-check"></i><b>4.1.6</b> <strong>Submit the main Snakemake job</strong></a></li>
<li class="chapter" data-level="4.1.7" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#bbc-maintained-reference-files"><i class="fa fa-check"></i><b>4.1.7</b> <strong>BBC-maintained reference files</strong></a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#symlink-pre-run-results-to-your-directory"><i class="fa fa-check"></i><b>4.2</b> <strong>Symlink pre-run results to your directory</strong></a></li>
<li class="chapter" data-level="4.3" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#quality-control"><i class="fa fa-check"></i><b>4.3</b> <strong>Quality control</strong></a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#multiqc-report"><i class="fa fa-check"></i><b>4.3.1</b> <strong>multiQC report</strong></a></li>
<li class="chapter" data-level="4.3.2" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#other-considerations"><i class="fa fa-check"></i><b>4.3.2</b> <strong>Other considerations</strong></a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#de-analysis-using-r-and-deseq2"><i class="fa fa-check"></i><b>4.4</b> <strong>DE analysis using R and DESeq2</strong></a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#create-an-r-project"><i class="fa fa-check"></i><b>4.4.1</b> <strong>Create an R project</strong></a></li>
<li class="chapter" data-level="4.4.2" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#exploring-the-data-using-isee"><i class="fa fa-check"></i><b>4.4.2</b> <strong>Exploring the data using iSEE</strong></a></li>
<li class="chapter" data-level="4.4.3" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#make-a-separate-folder-for-de-analysis-and-a-script-file-to-store-our-commands"><i class="fa fa-check"></i><b>4.4.3</b> <strong>Make a separate folder for DE analysis and a script file to store our commands</strong></a></li>
<li class="chapter" data-level="4.4.4" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#make-an-output-directory"><i class="fa fa-check"></i><b>4.4.4</b> <strong>Make an output directory</strong></a></li>
<li class="chapter" data-level="4.4.5" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#install-packages"><i class="fa fa-check"></i><b>4.4.5</b> <strong>Install packages</strong></a></li>
<li class="chapter" data-level="4.4.6" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#load-packages"><i class="fa fa-check"></i><b>4.4.6</b> <strong>Load packages</strong></a></li>
<li class="chapter" data-level="4.4.7" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#set-up-your-deseq-object"><i class="fa fa-check"></i><b>4.4.7</b> <strong>Set up your DESeq object</strong></a></li>
<li class="chapter" data-level="4.4.8" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#what-is-a-summarizedexperiment-object"><i class="fa fa-check"></i><b>4.4.8</b> <strong>What is a SummarizedExperiment object?</strong></a></li>
<li class="chapter" data-level="4.4.9" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#remove-genes-with-lowno-expression"><i class="fa fa-check"></i><b>4.4.9</b> <strong>Remove genes with low/no expression</strong></a></li>
<li class="chapter" data-level="4.4.10" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#different-normalization-approaches-for-different-biases"><i class="fa fa-check"></i><b>4.4.10</b> <strong>Different normalization approaches for different biases</strong></a></li>
<li class="chapter" data-level="4.4.11" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#run-the-de-workflow"><i class="fa fa-check"></i><b>4.4.11</b> <strong>Run the DE workflow</strong></a></li>
<li class="chapter" data-level="4.4.12" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#summarize-de-results"><i class="fa fa-check"></i><b>4.4.12</b> <strong>Summarize DE results</strong></a></li>
<li class="chapter" data-level="4.4.13" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#p-value-distribution"><i class="fa fa-check"></i><b>4.4.13</b> <strong>P value distribution</strong></a></li>
<li class="chapter" data-level="4.4.14" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#shrink-log-fold-changes-for-lowly-expressed-genes"><i class="fa fa-check"></i><b>4.4.14</b> <strong>Shrink log fold changes for lowly expressed genes</strong></a></li>
<li class="chapter" data-level="4.4.15" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#output-de-results"><i class="fa fa-check"></i><b>4.4.15</b> <strong>Output DE results</strong></a></li>
<li class="chapter" data-level="4.4.16" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#output-tables-with-log2-normalized-counts"><i class="fa fa-check"></i><b>4.4.16</b> <strong>Output tables with log2 normalized counts</strong></a></li>
<li class="chapter" data-level="4.4.17" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#why-should-we-transform-counts-for-visualizations-or-clustering"><i class="fa fa-check"></i><b>4.4.17</b> <strong>Why should we transform counts for visualizations or clustering?</strong></a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#volcano-plot"><i class="fa fa-check"></i><b>4.5</b> <strong>Volcano plot</strong></a></li>
<li class="chapter" data-level="4.6" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#make-a-new-summarizedexperiment-to-store-all-results"><i class="fa fa-check"></i><b>4.6</b> <strong>Make a new SummarizedExperiment to store all results</strong></a></li>
<li class="chapter" data-level="4.7" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#heatmap-of-top-de-genes"><i class="fa fa-check"></i><b>4.7</b> <strong>Heatmap of top DE genes</strong></a></li>
<li class="chapter" data-level="4.8" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#look-for-specific-genes"><i class="fa fa-check"></i><b>4.8</b> <strong>Look for specific genes</strong></a></li>
<li class="chapter" data-level="4.9" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#summarizedexperiment-exercises"><i class="fa fa-check"></i><b>4.9</b> <strong>SummarizedExperiment exercises</strong></a></li>
<li class="chapter" data-level="4.10" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#your-turn"><i class="fa fa-check"></i><b>4.10</b> <strong>Your turn</strong></a></li>
<li class="chapter" data-level="4.11" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#venn-diagrams"><i class="fa fa-check"></i><b>4.11</b> <strong>Venn diagrams</strong></a></li>
<li class="chapter" data-level="4.12" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#upset-plots"><i class="fa fa-check"></i><b>4.12</b> <strong>Upset plots</strong></a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="pathway-and-enrichment-analysis.html"><a href="pathway-and-enrichment-analysis.html"><i class="fa fa-check"></i><b>5</b> <strong>Pathway and enrichment analysis</strong></a>
<ul>
<li class="chapter" data-level="5.1" data-path="pathway-and-enrichment-analysis.html"><a href="pathway-and-enrichment-analysis.html#downstream-of-de-gene-analysis"><i class="fa fa-check"></i><b>5.1</b> <strong>Downstream of DE gene analysis</strong></a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="pathway-and-enrichment-analysis.html"><a href="pathway-and-enrichment-analysis.html#load-packages-1"><i class="fa fa-check"></i><b>5.1.1</b> <strong>load packages</strong></a></li>
<li class="chapter" data-level="5.1.2" data-path="pathway-and-enrichment-analysis.html"><a href="pathway-and-enrichment-analysis.html#import-previous-de-results"><i class="fa fa-check"></i><b>5.1.2</b> <strong>Import previous DE results</strong></a></li>
<li class="chapter" data-level="5.1.3" data-path="pathway-and-enrichment-analysis.html"><a href="pathway-and-enrichment-analysis.html#how-many-de-genes"><i class="fa fa-check"></i><b>5.1.3</b> <strong>How many DE genes</strong></a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="pathway-and-enrichment-analysis.html"><a href="pathway-and-enrichment-analysis.html#why-enrichment-and-pathway-analysis"><i class="fa fa-check"></i><b>5.2</b> <strong>Why enrichment and pathway analysis?</strong></a></li>
<li class="chapter" data-level="5.3" data-path="pathway-and-enrichment-analysis.html"><a href="pathway-and-enrichment-analysis.html#ora"><i class="fa fa-check"></i><b>5.3</b> <strong>ORA</strong></a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="pathway-and-enrichment-analysis.html"><a href="pathway-and-enrichment-analysis.html#plot-ora-results"><i class="fa fa-check"></i><b>5.3.1</b> <strong>plot ORA results</strong></a></li>
<li class="chapter" data-level="5.3.2" data-path="pathway-and-enrichment-analysis.html"><a href="pathway-and-enrichment-analysis.html#save-ora-results-to-a-tsv-file"><i class="fa fa-check"></i><b>5.3.2</b> <strong>save ORA results to a tsv file</strong></a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="pathway-and-enrichment-analysis.html"><a href="pathway-and-enrichment-analysis.html#gsea"><i class="fa fa-check"></i><b>5.4</b> <strong>GSEA</strong></a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="pathway-and-enrichment-analysis.html"><a href="pathway-and-enrichment-analysis.html#save-gsea-results"><i class="fa fa-check"></i><b>5.4.1</b> <strong>save gsea results</strong></a></li>
<li class="chapter" data-level="5.4.2" data-path="pathway-and-enrichment-analysis.html"><a href="pathway-and-enrichment-analysis.html#plot-gsea-results"><i class="fa fa-check"></i><b>5.4.2</b> <strong>plot GSEA results</strong></a></li>
<li class="chapter" data-level="5.4.3" data-path="pathway-and-enrichment-analysis.html"><a href="pathway-and-enrichment-analysis.html#plot-kegg-map"><i class="fa fa-check"></i><b>5.4.3</b> <strong>plot KEGG map</strong></a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="pathway-and-enrichment-analysis.html"><a href="pathway-and-enrichment-analysis.html#other-r-packages-and-tools"><i class="fa fa-check"></i><b>5.5</b> <strong>Other R packages and tools</strong></a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="appendices.html"><a href="appendices.html"><i class="fa fa-check"></i><b>6</b> <strong>Appendices</strong></a>
<ul>
<li class="chapter" data-level="6.1" data-path="appendices.html"><a href="appendices.html#other-workshop-files"><i class="fa fa-check"></i><b>6.1</b> <strong>Other workshop files</strong></a></li>
<li class="chapter" data-level="6.2" data-path="appendices.html"><a href="appendices.html#local-access-of-files-on-the-hpc"><i class="fa fa-check"></i><b>6.2</b> <strong>Local access of files on the HPC</strong></a></li>
<li class="chapter" data-level="6.3" data-path="appendices.html"><a href="appendices.html#bash-cheatsheet"><i class="fa fa-check"></i><b>6.3</b> <strong>Bash cheatsheet</strong></a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Introductory Bioinformatics for VAI Users</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="basic-rna-seq-analysis-from-fastq-files-to-de-genes" class="section level1 hasAnchor" number="4">
<h1><span class="header-section-number">Part 4</span> <strong>Basic RNA-Seq Analysis – From fastq files to DE genes</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#basic-rna-seq-analysis-from-fastq-files-to-de-genes" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="run-the-bbc-pipeline-to-align-the-reads" class="section level2 hasAnchor" number="4.1">
<h2><span class="header-section-number">4.1</span> <strong>Run the BBC pipeline to align the reads</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#run-the-bbc-pipeline-to-align-the-reads" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="the-workflow-as-a-graph" class="section level3 hasAnchor" number="4.1.1">
<h3><span class="header-section-number">4.1.1</span> <strong>The workflow as a graph</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#the-workflow-as-a-graph" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="float">
<img src="img/rulegraph.png" alt="The workflow." />
<div class="figcaption">The workflow.</div>
</div>
</div>
<div id="clone-the-github-repo" class="section level3 hasAnchor" number="4.1.2">
<h3><span class="header-section-number">4.1.2</span> <strong>Clone the Github repo</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#clone-the-github-repo" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>First, go to <a href="https://ondemand.vai.zone/">VAI OnDemand</a> and click on “Apps” &gt; “hpc Shell Access”.</p>
<p>Instructions for running the RNA-seq workflow are on the <a href="https://github.com/vari-bbc/rnaseq_workflow">Github README</a>, but here we will go through each step in more detail.</p>
<p>For a normal project, you would be working in your lab’s storage space, but for today we will work in the <code>hpctmp</code> folder.</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb70-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb70-1" tabindex="-1"></a><span class="bu">cd</span> /varidata/researchtemp/hpctmp/BBC_workshop_Oct2024_III/</span>
<span id="cb70-2"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb70-2" tabindex="-1"></a></span>
<span id="cb70-3"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb70-3" tabindex="-1"></a><span class="fu">mkdir</span> <span class="op">&lt;</span>username<span class="op">&gt;</span></span>
<span id="cb70-4"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb70-4" tabindex="-1"></a></span>
<span id="cb70-5"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb70-5" tabindex="-1"></a><span class="bu">cd</span> <span class="op">&lt;</span>username<span class="op">&gt;</span></span></code></pre></div>
<div class="sourceCode" id="cb71"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb71-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb71-1" tabindex="-1"></a><span class="fu">git</span> clone https://github.com/vari-bbc/rnaseq_workflow.git rnaseq_workflow_testrun</span></code></pre></div>
<p>The Github repository consisting of the RNA-seq workflow should now be downloaded in a folder named <code>rnaseq_workflow_testrun</code>.</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb72-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb72-1" tabindex="-1"></a><span class="bu">cd</span> rnaseq_workflow_testrun</span>
<span id="cb72-2"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb72-2" tabindex="-1"></a></span>
<span id="cb72-3"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb72-3" tabindex="-1"></a><span class="fu">ls</span></span></code></pre></div>
</div>
<div id="add-your-fastq-files-to-raw_data" class="section level3 hasAnchor" number="4.1.3">
<h3><span class="header-section-number">4.1.3</span> <strong>Add your fastq files to <code>raw_data/</code></strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#add-your-fastq-files-to-raw_data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Instead of making multiple copies of the same file, which can quickly use up your lab’s storage quota, we can use <strong>symbolic links</strong>. The sequence data that we will be using for this workshop are from the <a href="https://pubmed.ncbi.nlm.nih.gov/24926665/">airway dataset</a> referenced in the DESeq2 vignette. The gene counts can actually be downloaded as an <a href="https://bioconductor.org/packages/release/data/experiment/html/airway.html">R package</a>.</p>
<div class="sourceCode" id="cb73"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb73-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb73-1" tabindex="-1"></a><span class="fu">ls</span> ../../0_fastqs/</span></code></pre></div>
<div class="sourceCode" id="cb74"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb74-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb74-1" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-sr</span> ../../0_fastqs/<span class="pp">*</span> ./raw_data</span></code></pre></div>
<p>You can see where the symbolic links are pointing to using <code>ls -l</code>.</p>
<div class="sourceCode" id="cb75"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb75-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb75-1" tabindex="-1"></a><span class="fu">ls</span> <span class="at">-l</span> ./raw_data</span></code></pre></div>
</div>
<div id="fill-out-the-samplesheet" class="section level3 hasAnchor" number="4.1.4">
<h3><span class="header-section-number">4.1.4</span> <strong>Fill out the samplesheet</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#fill-out-the-samplesheet" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The samplesheet is a tab-delimited file within <code>config/samplesheet/</code> and is named <code>units.tsv</code>. The easiest way to fill this out is to run the helper script, <code>make_units_template.sh</code>, to generate a template, then edit using a text editor.</p>
<p>FOR THIS WORKSHOP, NO CHANGES NEED TO BE MADE.</p>
<div class="sourceCode" id="cb76"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb76-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb76-1" tabindex="-1"></a><span class="bu">cd</span> config/samplesheet/</span>
<span id="cb76-2"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb76-2" tabindex="-1"></a></span>
<span id="cb76-3"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb76-3" tabindex="-1"></a><span class="ex">./make_units_template.sh</span></span></code></pre></div>
<p>There should now be a file named ‘units_template.tsv’.</p>
<div class="sourceCode" id="cb77"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb77-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb77-1" tabindex="-1"></a><span class="fu">ls</span></span></code></pre></div>
<p>We can replace the ‘units.tsv’ with ‘units_template.tsv’.</p>
<div class="sourceCode" id="cb78"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb78-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb78-1" tabindex="-1"></a><span class="fu">cp</span> units_template.tsv units.tsv</span></code></pre></div>
<p>Use <code>cat</code> to look at the samplesheet.</p>
<div class="sourceCode" id="cb79"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb79-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb79-1" tabindex="-1"></a><span class="fu">cat</span> units.tsv</span></code></pre></div>
<p>The “group” column in <code>units.tsv</code> needs to be edited. Use the built-in editor in OnDemand to correct this column according to the values below. Tip: Within the OnDemand file explorer, click ‘Change directory’ and type <code>/varidata/researchtemp/hpctmp/BBC_workshop_Oct2024_III/&lt;username&gt;/rnaseq_workflow_testrun/config/samplesheet</code> to navigate directly to the correct folder.</p>
<table>
<thead>
<tr class="header">
<th>sample</th>
<th>group</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>SRR1039508</td>
<td>trt</td>
</tr>
<tr class="even">
<td>SRR1039509</td>
<td>untrt</td>
</tr>
<tr class="odd">
<td>SRR1039512</td>
<td>trt</td>
</tr>
<tr class="even">
<td>SRR1039513</td>
<td>untrt</td>
</tr>
<tr class="odd">
<td>SRR1039516</td>
<td>trt</td>
</tr>
<tr class="even">
<td>SRR1039517</td>
<td>untrt</td>
</tr>
<tr class="odd">
<td>SRR1039520</td>
<td>trt</td>
</tr>
<tr class="even">
<td>SRR1039521</td>
<td>untrt</td>
</tr>
</tbody>
</table>
<p>Go back to the base level of the project directory.</p>
<div class="sourceCode" id="cb80"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb80-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb80-1" tabindex="-1"></a><span class="bu">cd</span> ../..</span></code></pre></div>
<p>Make sure you are at <code>/varidata/researchtemp/hpctmp/BBC_workshop_Oct2024_III/&lt;username&gt;/rnaseq_workflow_testrun</code>.</p>
<div class="sourceCode" id="cb81"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb81-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb81-1" tabindex="-1"></a><span class="bu">pwd</span> </span></code></pre></div>
</div>
<div id="fill-out-the-config-file" class="section level3 hasAnchor" number="4.1.5">
<h3><span class="header-section-number">4.1.5</span> <strong>Fill out the config file</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#fill-out-the-config-file" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The config file is a <a href="https://en.wikipedia.org/wiki/YAML">YAML</a> file indicating the locations of reference files and also contains options for the workflow that you can turn off or turn on.</p>
<p>Typically, the main thing is to specify reference files corresponding to the species you have sequenced (human, mouse, or rat etc). For this workshop, we are dealing with human data so we will align to the hg38 reference.</p>
<p>Index files allow alignment algorithms to align reads to specific reference sequences.</p>
<p>FOR THIS WORKSHOP, NO CHANGES NEED TO BE MADE BECAUSE THE CORRECT INFORMATION HAVE ALREADY BEEN ENTERED.</p>
<div class="sourceCode" id="cb82"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb82-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb82-1" tabindex="-1"></a></span>
<span id="cb82-2"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb82-2" tabindex="-1"></a><span class="fu">cat</span> config/config.yaml</span></code></pre></div>
</div>
<div id="submit-the-main-snakemake-job" class="section level3 hasAnchor" number="4.1.6">
<h3><span class="header-section-number">4.1.6</span> <strong>Submit the main Snakemake job</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#submit-the-main-snakemake-job" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>If your lab has their own nodes that you wish to use, use <code>sbatch -p &lt;lab_name&gt; bin/run_snake.sh</code> instead of the command below.</p>
<div class="sourceCode" id="cb83"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb83-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb83-1" tabindex="-1"></a><span class="ex">sbatch</span> bin/run_snake.sh</span></code></pre></div>
</div>
<div id="bbc-maintained-reference-files" class="section level3 hasAnchor" number="4.1.7">
<h3><span class="header-section-number">4.1.7</span> <strong>BBC-maintained reference files</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#bbc-maintained-reference-files" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For future reference, the BBC downloads and maintains commonly used files and indexes for several model species. These files are version controlled to promote reproducibility in case you need to rerun an analysis or you want to run the exact same analysis on different datasets.</p>
<div class="sourceCode" id="cb84"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb84-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb84-1" tabindex="-1"></a><span class="fu">ls</span> /varidata/research/projects/bbc/versioned_references/</span></code></pre></div>
<pre><code>## 2021-04-09_12.44.55_v1
## 2021-07-02_11.23.48_v3
## 2021-07-13_09.57.31_v4
## 2021-08-02_20.40.24_v5
## 2021-08-10_11.12.27_v6
## 2021-10-25_15.31.38_v7
## 2022-01-26_13.39.07_v8
## 2022-03-08_14.47.50_v9
## 2022-10-06_14.25.40_v10
## 2022-12-19_13.27.04_v11
## 2023-05-03_15.28.41_v12
## 2023-10-04_10.07.51_v13
## 2023-10-06_11.13.48_v14
## 2023-11-08_15.45.32_v15
## 2023-11-09_09.38.05_v16
## 2024-10-31_10.56.03_v17
## 2025-02-04_16.03.25_v18
## 2025-04-30_23.47.46_v19
## 2025-05-12_16.16.50_v20
## latest
## regarding_v2.txt</code></pre>
<div class="sourceCode" id="cb86"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb86-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb86-1" tabindex="-1"></a><span class="fu">ls</span> /varidata/research/projects/bbc/versioned_references/2023-05-03_15.28.41_v12/data/</span></code></pre></div>
<pre><code>## GRCz11
## c.elegans-WBcel235
## dm6_BDGP6.28.100
## e.coli-K12-mg1665_ensembl
## hg19_gencode
## hg19_gencode_plus_ERCC92
## hg19_gencode_plus_viruses
## hg38_gencode
## mm10_gencode
## mm10_gencode_plus_ERCC92
## mm10_gencode_plus_e.coli-K12-mg1665_ensembl
## mm10_gencode_plus_viruses_and_cfmedips
## mm11_gencode
## mm9_ucsc
## mm_BALB_CJ
## mm_BALB_CJ_plus_viruses_and_cfmedips
## mm_FVB_NJ
## mm_FVB_NJ_plus_viruses_and_cfmedips
## rnor6_ensembl</code></pre>
<p>The source of these files can be found in the <code>species.tsv</code> file.</p>
<div class="sourceCode" id="cb88"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb88-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb88-1" tabindex="-1"></a><span class="fu">cat</span> /varidata/research/projects/bbc/versioned_references/2023-05-03_15.28.41_v12/bin/species.tsv  <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-f2-3</span></span></code></pre></div>
<pre><code>## id   genome_fasta
## mm10_gencode ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M24/GRCm38.primary_assembly.genome.fa.gz
## hg38_gencode ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_33/GRCh38.primary_assembly.genome.fa.gz
## e.coli-K12-mg1665_ensembl    ftp://ftp.ensemblgenomes.org/pub/release-47/bacteria//fasta/bacteria_0_collection/escherichia_coli_str_k_12_substr_mg1655/dna/Escherichia_coli_str_k_12_substr_mg1655.ASM584v2.dna.chromosome.Chromosome.fa.gz
## hg19_gencode ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_36/GRCh37_mapping/GRCh37.primary_assembly.genome.fa.gz
## dm6_BDGP6.28.100 ftp://ftp.ensembl.org/pub/release-100/fasta/drosophila_melanogaster/dna/Drosophila_melanogaster.BDGP6.28.dna.toplevel.fa.gz
## c.elegans-WBcel235   http://ftp.ensembl.org/pub/release-103/fasta/caenorhabditis_elegans/dna/Caenorhabditis_elegans.WBcel235.dna.toplevel.fa.gz
## mm_BALB_CJ   http://ftp.ensembl.org/pub/release-104/fasta/mus_musculus_balbcj/dna/Mus_musculus_balbcj.BALB_cJ_v1.dna.toplevel.fa.gz
## mm_FVB_NJ    http://ftp.ensembl.org/pub/release-104/fasta/mus_musculus_fvbnj/dna/Mus_musculus_fvbnj.FVB_NJ_v1.dna.toplevel.fa.gz
## rnor6_ensembl    http://ftp.ensembl.org/pub/release-104/fasta/rattus_norvegicus/dna/Rattus_norvegicus.Rnor_6.0.dna.toplevel.fa.gz
## GRCz11   https://ftp.ensembl.org/pub/release-108/fasta/danio_rerio/dna/Danio_rerio.GRCz11.dna.toplevel.fa.gz
## mm11_gencode https://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M31/GRCm39.primary_assembly.genome.fa.gz
## mm9_ucsc ftp://hgdownload.cse.ucsc.edu/goldenPath/mm9/bigZips/mm9.fa.gz</code></pre>
<p>The versions of the software used to generate the index files can be found in the <code>config.yaml</code> file. Sometimes index files are not backwards-compatible, meaning index files generated by a newer version of a tool cannot be used by an older version of the tool.</p>
<div class="sourceCode" id="cb90"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb90-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb90-1" tabindex="-1"></a><span class="fu">cat</span> /varidata/research/projects/bbc/versioned_references/2023-05-03_15.28.41_v12/bin/config.yaml</span></code></pre></div>
<pre><code>## # This is the directory that the rsync rule will use as the source directory. It shuold be the run directory this workflow.
## sourceDir: &quot;/varidata/research/projects/bbc/research/prep_bbc_shared_current&quot;
## 
## # This is where the timestamped directories and the &#39;latest&#39; symlink will be created.
## timestamp_dir: &quot;/varidata/research/projects/bbc/versioned_references/&quot;
## 
## # Environment modules used.
## # Specify all environment modules here instead of within the Snakefile itself to make it easy to look over quickly.
## # When aligning using the index files created by this workflow, use the same version aligner as listed here to ensure compatibility.
## samtools:   &quot;bbc/samtools/samtools-1.9&quot;
## picard:     &quot;bbc/picard/picard-2.21.4-SNAPSHOT&quot;
## STAR:       &quot;bbc/STAR/STAR-2.7.8a&quot;
## biscuit:    &quot;bbc/biscuit/biscuit_1_0_1&quot;
## bwa:        &quot;bbc/bwa/bwa-0.7.17&quot;
## bowtie2:    &quot;bbc/bowtie2/bowtie2-2.4.1&quot;
## python3:    &quot;bbc/python3/python-3.8.1&quot;
## bismark:    &quot;bbc/bismark/bismark-0.23.0&quot;
## kb-python:  &quot;bbc/kb-python/kb-python-0.24.4&quot;
## parallel:   &quot;bbc/parallel/parallel-20191122&quot;
## gsutil:     &quot;bbc/gsutil/gsutil-4.52&quot;
## seqtk:      &quot;bbc/seqtk/seqtk-1.3-r115-dirty&quot;
## salmon:     &quot;bbc/salmon/salmon-1.4.0&quot;
## kallisto:   &quot;bbc/kallisto/kallisto-0.46.1&quot;</code></pre>
</div>
</div>
<div id="symlink-pre-run-results-to-your-directory" class="section level2 hasAnchor" number="4.2">
<h2><span class="header-section-number">4.2</span> <strong>Symlink pre-run results to your directory</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#symlink-pre-run-results-to-your-directory" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Because the workflow will take a while (up to several hours) to run, we have run the workflow for you beforehand so that you don’t have to wait for your workflow to finish running.</p>
<p>First, go to the base level of your “<username>” directory.</p>
<div class="sourceCode" id="cb92"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb92-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb92-1" tabindex="-1"></a><span class="bu">cd</span> /varidata/researchtemp/hpctmp/BBC_workshop_Oct2024_III/<span class="op">&lt;</span>username<span class="op">&gt;</span></span></code></pre></div>
<p>Then create the symlink.</p>
<div class="sourceCode" id="cb93"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb93-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb93-1" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-s</span> ../kin.lau/rnaseq_workflow/ .</span></code></pre></div>
</div>
<div id="quality-control" class="section level2 hasAnchor" number="4.3">
<h2><span class="header-section-number">4.3</span> <strong>Quality control</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#quality-control" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>It’s important to look through QC metrics to ensure that the data is likely to produce meaningful results. Ideally, you don’t want to waste time trying to interpret bad data.</p>
<div id="multiqc-report" class="section level3 hasAnchor" number="4.3.1">
<h3><span class="header-section-number">4.3.1</span> <strong>multiQC report</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#multiqc-report" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Use the OnDemand file explorer to download and open <code>/varidata/researchtemp/hpctmp/BBC_workshop_Oct2024_III/&lt;username&gt;/rnaseq_workflow/results/multiqc/multiqc_report.html</code>. <em>Remember</em> that within the OnDemand file explorer, you can click ‘Change directory’ to navigate directly to the enclosing directory.</p>
<ul>
<li>Alignment rate?</li>
<li>Higher duplication rate can be normal for RNA-seq.</li>
<li>Strandedness; In our workflow, this is automatically inferred using Salmon.</li>
<li>Any contamination? Check FastqScreen results.</li>
</ul>
</div>
<div id="other-considerations" class="section level3 hasAnchor" number="4.3.2">
<h3><span class="header-section-number">4.3.2</span> <strong>Other considerations</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#other-considerations" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>Not a bad idea to look quickly at the alignments in IGV. Any sign of gDNA contamination?</li>
<li>Mutations in specific genotypes? Is it exonic? Can do a quick check in IGV.</li>
<li>How many genes expressed? Easy to check in R.</li>
<li>Marker genes? Can do a quick check in iSEE if you expect certain expression patterns between groups.</li>
<li>Any genes supposed to knocked out or knocked down? Use iSEE.</li>
<li>Does the PCA show clustering? Use iSEE.</li>
</ul>
</div>
</div>
<div id="de-analysis-using-r-and-deseq2" class="section level2 hasAnchor" number="4.4">
<h2><span class="header-section-number">4.4</span> <strong>DE analysis using R and DESeq2</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#de-analysis-using-r-and-deseq2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="create-an-r-project" class="section level3 hasAnchor" number="4.4.1">
<h3><span class="header-section-number">4.4.1</span> <strong>Create an R project</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#create-an-r-project" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p>Please go to <a href="https://ondemand.vai.zone/">VAI OnDemand</a>.</p></li>
<li><p>Click on RStudio Server, and create a new RStudio session. Please make sure you are loading the correct R module for this workshop, <code>bbc2/R/alt/R-4.4.0-setR_LIBS_USER</code>, which you can select from the drop down menu. Type “short” in the “Partition” field (if you encounter issues, you can also try “long”). Set CPU as 1, memory as 16GB, and time as 6 hours. Click “Launch”.</p></li>
<li><p>In the Rstudio window, click “File” in the upper left corner, click on “New Project” &gt; “New Directory” &gt; “New Project”. Then, click on “Browse…”, you can click on triple dot horizontal button and type <code>/varidata/researchtemp/hpctmp/BBC_workshop_Oct2024_III/&lt;username&gt;</code>. Click “choose”. Next, you will go back to the project wizard, and type “workshop_2024” in the “Directory name” field. Click “Create Project”. You will be the newly created “workshop_2024” folder, which will contain a new file named “workshop_2024.Rproj”, which Rstudio uses to store details about your R project.</p></li>
</ol>
<div id="why-create-r-projects" class="section level4 hasAnchor" number="4.4.1.1">
<h4><span class="header-section-number">4.4.1.1</span> Why create R projects?<a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#why-create-r-projects" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><strong>Creating an R Project in RStudio is recommended for several reasons:</strong></p>
<ol style="list-style-type: decimal">
<li><p>A R Project creates a dedicated working directory for your project files, including R scripts, data files, figures, and output. This helps in organizing your files that are related to a specific project.</p></li>
<li><p>Automatic working directory setup. When you open an R Project, RStudio automatically sets the working directory to the project folder.</p></li>
<li><p>Reproducibility. With a project-based setup, the code and files related to a project stay together, making it easier to reproduce results.</p></li>
<li><p>Version control integration and package development support.</p></li>
</ol>
</div>
</div>
<div id="exploring-the-data-using-isee" class="section level3 hasAnchor" number="4.4.2">
<h3><span class="header-section-number">4.4.2</span> <strong>Exploring the data using iSEE</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#exploring-the-data-using-isee" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The <a href="https://www.bioconductor.org/packages/release/bioc/html/iSEE.html">iSEE</a> app allows us to explore our data interactively, making changes on-the-fly. It is more convenient to mount the HPC storage to your computer and run iSEE locally, but for simplicity we will follow the steps below to run iSEE via OnDemand Rstudio.</p>
<p>Go to the terminal. Go to your newly created <code>workshop_2024</code> R project directory, then copy <code>app.R</code> and <code>sce.rds</code> from <code>rnaseq_workflow/iSEE/</code> to there.</p>
<div class="sourceCode" id="cb94"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb94-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb94-1" tabindex="-1"></a><span class="bu">cd</span> /varidata/researchtemp/hpctmp/BBC_workshop_Oct2024_III/<span class="op">&lt;</span>username<span class="op">&gt;</span>/workshop_2024</span>
<span id="cb94-2"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb94-2" tabindex="-1"></a></span>
<span id="cb94-3"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb94-3" tabindex="-1"></a><span class="fu">cp</span> ../rnaseq_workflow/iSEE/<span class="pp">*</span> .</span></code></pre></div>
<p>To run the iSEE app, go back to OnDemand Rstudio, open the <code>app.R</code> file and click “Source”.</p>
<div class="float">
<img src="img/iSEE_screenshot.png" alt="iSEE screenshot." />
<div class="figcaption">iSEE screenshot.</div>
</div>
</div>
<div id="make-a-separate-folder-for-de-analysis-and-a-script-file-to-store-our-commands" class="section level3 hasAnchor" number="4.4.3">
<h3><span class="header-section-number">4.4.3</span> <strong>Make a separate folder for DE analysis and a script file to store our commands</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#make-a-separate-folder-for-de-analysis-and-a-script-file-to-store-our-commands" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p>We will create a new folder for the differential expression analysis of this workshop. Click “New Folder” and enter “DE_genes” for the folder name.</p></li>
<li><p>Create an R script file to record the code that you will use for this part of the workshop today. Click on the “DE_genes” folder to enter it. Click “New Blank File” &gt; “R script”. Type in “deseq2.R” for the filename and click “OK”. Unless otherwise noted, run all following code by copying and pasting into “deseq2.R” then highlighting the commands you want to run and typing either Cmd + Enter (Mac) or Ctrl + Enter (Windows).</p></li>
</ol>
</div>
<div id="make-an-output-directory" class="section level3 hasAnchor" number="4.4.4">
<h3><span class="header-section-number">4.4.4</span> <strong>Make an output directory</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#make-an-output-directory" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To ensure it is clear which files are produced by a given script, it is good practice to make a unique folder for storing each script’s output.</p>
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb95-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb95-1" tabindex="-1"></a>outdir <span class="ot">&lt;-</span> <span class="st">&quot;./DE_genes/deseq2_out_files/&quot;</span></span>
<span id="cb95-2"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb95-2" tabindex="-1"></a></span>
<span id="cb95-3"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb95-3" tabindex="-1"></a><span class="fu">dir.create</span>(outdir, <span class="at">recursive=</span><span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div id="install-packages" class="section level3 hasAnchor" number="4.4.5">
<h3><span class="header-section-number">4.4.5</span> <strong>Install packages</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#install-packages" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For this workshop, we ran the Rstudio job with the <code>bbc2/R/alt/R-4.4.0-setR_LIBS_USER</code> module. This allows you to use R packages that we have installed beforehand. For your actual work, we encourage you to run Rstudio jobs with the <code>bbc2/R/R-4.4.0</code> module which will allows you to make and maintain your own R package library, avoiding permission issues when updating packages or installing new ones.</p>
<p>While CRAN packages can be installed with <code>install.packages()</code>, <a href="https://bioconductor.org/">Bioconductor</a> packages have to be installed with the <code>install()</code> function in the <a href="https://cran.r-project.org/web/packages/BiocManager/vignettes/BiocManager.html">BiocManager package</a>.</p>
<div class="sourceCode" id="cb96"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb96-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb96-1" tabindex="-1"></a><span class="co"># AS EXPLAINED ABOVE, NO NEED TO RUN THESE FOR THE WORKSHOP</span></span>
<span id="cb96-2"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb96-2" tabindex="-1"></a><span class="co"># if (!require(&quot;BiocManager&quot;, quietly = TRUE))</span></span>
<span id="cb96-3"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb96-3" tabindex="-1"></a><span class="co">#     install.packages(&quot;BiocManager&quot;)</span></span>
<span id="cb96-4"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb96-4" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb96-5"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb96-5" tabindex="-1"></a><span class="co"># BiocManager::install(c(&quot;dplyr&quot;,&quot;stringr&quot;,&quot;ggplot2&quot;,&quot;readr&quot;,&quot;ggrepel&quot;,&quot;ComplexHeatmap&quot;,&quot;DESeq2&quot;,&quot;iSEE&quot;))</span></span></code></pre></div>
</div>
<div id="load-packages" class="section level3 hasAnchor" number="4.4.6">
<h3><span class="header-section-number">4.4.6</span> <strong>Load packages</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#load-packages" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>In order to let R know where to look for the functions we will use, we need to load them using the <code>library()</code> function. The <code>suppressPackageStartupMessages()</code> function calls here are simply to mute the automatic start-up messages of some of these packages to avoid cluttering what we are looking at.</p>
<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb97-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb97-1" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(dplyr))</span>
<span id="cb97-2"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb97-2" tabindex="-1"></a><span class="fu">library</span>(stringr)</span>
<span id="cb97-3"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb97-3" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb97-4"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb97-4" tabindex="-1"></a><span class="fu">library</span>(readr)</span>
<span id="cb97-5"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb97-5" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb97-6"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb97-6" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(ComplexHeatmap))</span>
<span id="cb97-7"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb97-7" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(DESeq2))</span></code></pre></div>
</div>
<div id="set-up-your-deseq-object" class="section level3 hasAnchor" number="4.4.7">
<h3><span class="header-section-number">4.4.7</span> <strong>Set up your DESeq object</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#set-up-your-deseq-object" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The BBC Snakemake workflow collates all the gene counts into an R object saved in the <code>results/SummarizedExperiment/SummarizedExperiment.rds</code> file. The <code>readRDS</code> function loads that object into our environment.</p>
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb98-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb98-1" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;../rnaseq_workflow/results/SummarizedExperiment/SummarizedExperiment.rds&quot;</span>)</span>
<span id="cb98-2"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb98-2" tabindex="-1"></a></span>
<span id="cb98-3"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb98-3" tabindex="-1"></a><span class="co"># What is type of object is this?</span></span>
<span id="cb98-4"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb98-4" tabindex="-1"></a><span class="fu">class</span>(se)</span></code></pre></div>
</div>
<div id="what-is-a-summarizedexperiment-object" class="section level3 hasAnchor" number="4.4.8">
<h3><span class="header-section-number">4.4.8</span> <strong>What is a SummarizedExperiment object?</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#what-is-a-summarizedexperiment-object" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>A <a href="https://bioconductor.org/packages/release/bioc/vignettes/SummarizedExperiment/inst/doc/SummarizedExperiment.html">SummarizedExperiment</a> object allows us to store and manipulate sample meta data, gene meta data and gene counts in separate, coordinated dataframes and matrices. Subsetting and reordering samples or genes can be done using the familiar <code>object[row, column]</code> R syntax, where the genes are the rows and the samples are the columns (<code>object[genes, samples]</code>). Later on, we will go through some exercises to further demonstrate the usefulness of “SummarizedExperiment” objects.</p>
<div class="float">
<img src="img/SummarizedExperiment.png" alt="SummarizedExperiment" />
<div class="figcaption">SummarizedExperiment</div>
</div>
<p>Let’s take a look to see what assays are stored in the SummarizedExperiment object. Note that DESeq2 requires the first assay to be named “counts” and assumes it contains the raw counts.</p>
<div class="sourceCode" id="cb99"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb99-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb99-1" tabindex="-1"></a><span class="fu">assayNames</span>(se)</span></code></pre></div>
<pre><code>## [1] &quot;counts&quot; &quot;tpms&quot;   &quot;vst&quot;</code></pre>
<div class="sourceCode" id="cb101"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb101-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb101-1" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">assayNames</span>(se)[<span class="dv">1</span>] <span class="sc">==</span> <span class="st">&quot;counts&quot;</span>)</span></code></pre></div>
<p>Let’s take a look at the sample meta data in the “se” object. The BBC snakemake workflow automatically adds the samplesheet metadata into the “se” object.</p>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb102-1" tabindex="-1"></a><span class="fu">colData</span>(se)</span></code></pre></div>
<pre><code>## DataFrame with 8 rows and 3 columns
##                 sample       group        RG
##            &lt;character&gt; &lt;character&gt; &lt;logical&gt;
## SRR1039508  SRR1039508         trt        NA
## SRR1039512  SRR1039512         trt        NA
## SRR1039516  SRR1039516         trt        NA
## SRR1039520  SRR1039520         trt        NA
## SRR1039509  SRR1039509       untrt        NA
## SRR1039513  SRR1039513       untrt        NA
## SRR1039517  SRR1039517       untrt        NA
## SRR1039521  SRR1039521       untrt        NA</code></pre>
<p>Similarly, the gene meta data can be used to store alternate gene names. These will come in handy for downstream visualizations or pathway analyses.</p>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb104-1" tabindex="-1"></a><span class="fu">rowData</span>(se)</span></code></pre></div>
<pre><code>## DataFrame with 60721 rows and 4 columns
##                         Symbol              Uniq_syms      entrez
##                    &lt;character&gt;            &lt;character&gt; &lt;character&gt;
## ENSG00000223972.5           NA      ENSG00000223972.5          NA
## ENSG00000227232.5           NA      ENSG00000227232.5          NA
## ENSG00000278267.1    MIR6859-2 MIR6859-2_ENSG000002..   102465909
## ENSG00000243485.5           NA      ENSG00000243485.5          NA
## ENSG00000284332.1    MIR1302-2              MIR1302-2   100302278
## ...                        ...                    ...         ...
## ENSG00000276017.1           NA      ENSG00000276017.1          NA
## ENSG00000278817.1        DGCR6 DGCR6_ENSG0000027881..        8214
## ENSG00000277196.4        PRODH PRODH_ENSG0000027719..        5625
## ENSG00000278625.1 LOC124905334           LOC124905334   124905334
## ENSG00000277374.1           NA      ENSG00000277374.1          NA
##                                Gene_name
##                              &lt;character&gt;
## ENSG00000223972.5                     NA
## ENSG00000227232.5                     NA
## ENSG00000278267.1        microRNA 6859-2
## ENSG00000243485.5                     NA
## ENSG00000284332.1        microRNA 1302-2
## ...                                  ...
## ENSG00000276017.1                     NA
## ENSG00000278817.1 DiGeorge syndrome cr..
## ENSG00000277196.4 proline dehydrogenas..
## ENSG00000278625.1    U6 spliceosomal RNA
## ENSG00000277374.1                     NA</code></pre>
<p>The counts and the meta data need to be stored inside a DESeq2 object called a ‘DESeqDataSet’, which is also a ‘SummarizedExperiment’ so ‘SummarizedExperiment’ functions will work on it. We specify that each gene will be fit with a model design of ‘~ group’.</p>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb106-1" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSet</span>(se, <span class="at">design =</span> <span class="sc">~</span> group)</span></code></pre></div>
<pre><code>## converting counts to integer mode</code></pre>
<pre><code>## Warning in DESeqDataSet(se, design = ~group): some variables in design formula
## are characters, converting to factors</code></pre>
</div>
<div id="remove-genes-with-lowno-expression" class="section level3 hasAnchor" number="4.4.9">
<h3><span class="header-section-number">4.4.9</span> <strong>Remove genes with low/no expression</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#remove-genes-with-lowno-expression" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We cannot do meaningful analyses of genes with very low counts. This will also speed up the analysis.</p>
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb109-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb109-1" tabindex="-1"></a><span class="co"># prefilter genes, keeping only genes with 10 or more total read counts across samples</span></span>
<span id="cb109-2"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb109-2" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(dds)) <span class="sc">&gt;=</span> <span class="dv">10</span> <span class="co"># counts() is the same as assay(dds, &quot;counts&quot;)</span></span>
<span id="cb109-3"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb109-3" tabindex="-1"></a></span>
<span id="cb109-4"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb109-4" tabindex="-1"></a><span class="fu">message</span>(<span class="fu">str_glue</span>(<span class="st">&quot;Keeping {sum(keep)} genes.&quot;</span>))</span></code></pre></div>
<pre><code>## Keeping 21485 genes.</code></pre>
<div class="sourceCode" id="cb111"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb111-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb111-1" tabindex="-1"></a>dds <span class="ot">&lt;-</span> dds[keep, ]</span></code></pre></div>
</div>
<div id="different-normalization-approaches-for-different-biases" class="section level3 hasAnchor" number="4.4.10">
<h3><span class="header-section-number">4.4.10</span> <strong>Different normalization approaches for different biases</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#different-normalization-approaches-for-different-biases" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><a href="https://vanandelinstitute-my.sharepoint.com/:b:/g/personal/kin_lau_vai_org/EcruvwL-OrBIvCzXZ7HMPlcBo65fu0pucrivMmCwzM98dA?e=yCkfTa">Types of biases in RNA-seq</a></p>
</div>
<div id="run-the-de-workflow" class="section level3 hasAnchor" number="4.4.11">
<h3><span class="header-section-number">4.4.11</span> <strong>Run the DE workflow</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#run-the-de-workflow" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The <a href="https://www.rdocumentation.org/packages/DESeq2/versions/1.12.3/topics/DESeq">DESeq</a> function is a convenience function from DESeq2 that estimates size factors (normalization) and fits negative binomial GLMs.</p>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb112-1" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span></code></pre></div>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<div class="sourceCode" id="cb119"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb119-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb119-1" tabindex="-1"></a><span class="fu">message</span>(<span class="fu">paste0</span>(<span class="st">&quot;Coefficient names are: &quot;</span>, <span class="fu">paste</span>(<span class="fu">resultsNames</span>(dds), <span class="at">collapse =</span> <span class="st">&quot; &quot;</span>)))</span></code></pre></div>
<pre><code>## Coefficient names are: Intercept group_untrt_vs_trt</code></pre>
<p>After the models are fitted, we can test specific pairs of groups for differential expression. For DESeq2, it is recommended to provide the significance cutoff that you wish to use as it affects the multiple testing correction procedure (see <a href="https://www.rdocumentation.org/packages/DESeq2/versions/1.12.3/topics/results">docs</a>). Here we specify a significance cutoff of 0.05.</p>
<div class="sourceCode" id="cb121"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb121-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb121-1" tabindex="-1"></a>contrast <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;group&quot;</span>, <span class="st">&quot;trt&quot;</span>, <span class="st">&quot;untrt&quot;</span>)</span>
<span id="cb121-2"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb121-2" tabindex="-1"></a>fdr_cutoff <span class="ot">&lt;-</span> <span class="fl">0.05</span></span>
<span id="cb121-3"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb121-3" tabindex="-1"></a></span>
<span id="cb121-4"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb121-4" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">results</span>(dds, <span class="at">contrast=</span>contrast, <span class="at">alpha=</span>fdr_cutoff)</span>
<span id="cb121-5"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb121-5" tabindex="-1"></a>res <span class="ot">&lt;-</span> res[<span class="fu">order</span>(res<span class="sc">$</span>pvalue), ]</span></code></pre></div>
</div>
<div id="summarize-de-results" class="section level3 hasAnchor" number="4.4.12">
<h3><span class="header-section-number">4.4.12</span> <strong>Summarize DE results</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#summarize-de-results" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Let’s see how many genes were significantly up or down regulated.</p>
<div class="sourceCode" id="cb122"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb122-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb122-1" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(res)</span>
<span id="cb122-2"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb122-2" tabindex="-1"></a><span class="fu">data.frame</span>(</span>
<span id="cb122-3"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb122-3" tabindex="-1"></a>  <span class="at">UP=</span><span class="fu">sum</span>(df<span class="sc">$</span>padj <span class="sc">&lt;=</span> fdr_cutoff <span class="sc">&amp;</span> df<span class="sc">$</span>log2FoldChange <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb122-4"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb122-4" tabindex="-1"></a>  <span class="at">DWN=</span><span class="fu">sum</span>(df<span class="sc">$</span>padj <span class="sc">&lt;=</span> fdr_cutoff <span class="sc">&amp;</span> df<span class="sc">$</span>log2FoldChange <span class="sc">&lt;</span> <span class="dv">0</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb122-5"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb122-5" tabindex="-1"></a>  <span class="at">Tested=</span><span class="fu">sum</span>(<span class="sc">!</span><span class="fu">is.na</span>(df<span class="sc">$</span>padj))</span>
<span id="cb122-6"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb122-6" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>##     UP  DWN Tested
## 1 1158 1534  18105</code></pre>
</div>
<div id="p-value-distribution" class="section level3 hasAnchor" number="4.4.13">
<h3><span class="header-section-number">4.4.13</span> <strong>P value distribution</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#p-value-distribution" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Ideally, we will see an anti-conservative (if there are many DE genes) or uniform pattern (not many DE genes). See <a href="http://varianceexplained.org/statistics/interpreting-pvalue-histogram/">here</a> for more details about how to interpret these.</p>
<div class="sourceCode" id="cb124"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb124-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb124-1" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> df <span class="sc">%&gt;%</span></span>
<span id="cb124-2"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb124-2" tabindex="-1"></a>         dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(pvalue)),</span>
<span id="cb124-3"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb124-3" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> pvalue)) <span class="sc">+</span></span>
<span id="cb124-4"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb124-4" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;gray55&quot;</span>,</span>
<span id="cb124-5"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb124-5" tabindex="-1"></a>                 <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="fl">0.05</span>)) <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">plot.title=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">10</span>))</span></code></pre></div>
<p><img src="03_rnaseq_workflow_files/figure-html/pval-1.png" width="384" /></p>
</div>
<div id="shrink-log-fold-changes-for-lowly-expressed-genes" class="section level3 hasAnchor" number="4.4.14">
<h3><span class="header-section-number">4.4.14</span> <strong>Shrink log fold changes for lowly expressed genes</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#shrink-log-fold-changes-for-lowly-expressed-genes" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This step does not affect which genes are statistically DE, but we perform this to obtain more reliable estimates of the log fold changes for visualizations or for ranking genes (e.g. GSEA).</p>
<div class="sourceCode" id="cb125"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb125-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb125-1" tabindex="-1"></a>lfc_shrink <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(dds, <span class="at">contrast=</span>contrast, <span class="at">type=</span><span class="st">&quot;ashr&quot;</span>)</span></code></pre></div>
<pre><code>## using &#39;ashr&#39; for LFC shrinkage. If used in published research, please cite:
##     Stephens, M. (2016) False discovery rates: a new deal. Biostatistics, 18:2.
##     https://doi.org/10.1093/biostatistics/kxw041</code></pre>
<div class="sourceCode" id="cb127"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb127-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb127-1" tabindex="-1"></a>lfc_shrink <span class="ot">&lt;-</span> lfc_shrink[<span class="fu">order</span>(lfc_shrink<span class="sc">$</span>pvalue), ]</span></code></pre></div>
<p>Let’s visualize the effect of shrinking the LFCs using MA plots.</p>
<div class="sourceCode" id="cb128"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb128-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb128-1" tabindex="-1"></a>DESeq2<span class="sc">::</span><span class="fu">plotMA</span>(res, <span class="at">main=</span><span class="st">&quot;Default LFC&quot;</span>)</span></code></pre></div>
<p><img src="03_rnaseq_workflow_files/figure-html/ma_plots-1.png" width="480" /></p>
<div class="sourceCode" id="cb129"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb129-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb129-1" tabindex="-1"></a>DESeq2<span class="sc">::</span><span class="fu">plotMA</span>(lfc_shrink, <span class="at">main=</span><span class="st">&quot;Shrunken LFC&quot;</span>)</span></code></pre></div>
<p><img src="03_rnaseq_workflow_files/figure-html/ma_plots_shrunken-1.png" width="480" /></p>
</div>
<div id="output-de-results" class="section level3 hasAnchor" number="4.4.15">
<h3><span class="header-section-number">4.4.15</span> <strong>Output DE results</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#output-de-results" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Here, we merge the different gene name columns to the DE results and output to a tab-delimited file, which can be opened in Excel for manual perusal.</p>
<p>NOTE: You will use this file for the next section of the workshop about pathway analysis.</p>
<div class="sourceCode" id="cb130"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb130-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb130-1" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">as.data.frame</span>(<span class="fu">rowData</span>(dds)[<span class="fu">rownames</span>(lfc_shrink), <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>]),</span>
<span id="cb130-2"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb130-2" tabindex="-1"></a>            <span class="fu">as.data.frame</span>(lfc_shrink)) <span class="sc">%&gt;%</span></span>
<span id="cb130-3"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb130-3" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;ens_gene&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb131"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb131-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb131-1" tabindex="-1"></a><span class="fu">write_tsv</span>(df, <span class="fu">file.path</span>(outdir, <span class="st">&quot;de_res.tsv&quot;</span>))</span></code></pre></div>
</div>
<div id="output-tables-with-log2-normalized-counts" class="section level3 hasAnchor" number="4.4.16">
<h3><span class="header-section-number">4.4.16</span> <strong>Output tables with log2 normalized counts</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#output-tables-with-log2-normalized-counts" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For the log2 normalized counts, we commonly use the variance stabilized transformation (<a href="https://rdrr.io/bioc/DESeq2/man/varianceStabilizingTransformation.html">VST</a>). These values can be used for heatmaps, clustering or other downstream applications.</p>
<div class="sourceCode" id="cb132"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb132-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb132-1" tabindex="-1"></a>vsd <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds, <span class="at">blind=</span><span class="cn">FALSE</span>)</span>
<span id="cb132-2"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb132-2" tabindex="-1"></a></span>
<span id="cb132-3"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb132-3" tabindex="-1"></a>vst_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">cbind</span>(<span class="fu">rowData</span>(vsd)[, <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>], <span class="fu">assay</span>(vsd))) <span class="sc">%&gt;%</span></span>
<span id="cb132-4"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb132-4" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;ens_gene&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb133"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb133-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb133-1" tabindex="-1"></a><span class="fu">write_tsv</span>(vst_df, <span class="fu">file.path</span>(outdir, <span class="st">&quot;vst.tsv&quot;</span>))</span></code></pre></div>
</div>
<div id="why-should-we-transform-counts-for-visualizations-or-clustering" class="section level3 hasAnchor" number="4.4.17">
<h3><span class="header-section-number">4.4.17</span> <strong>Why should we transform counts for visualizations or clustering?</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#why-should-we-transform-counts-for-visualizations-or-clustering" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Biological factors can increase gene count variation, but this can be muddied by dependence of the variance on the magnitude of the mean. Below, we can visualize how different transformations impact the mean-variance relationship. Here we also reiterate that these transformations are only for analyses downstream of DE analysis; the statistical framework in DESeq2 (and edgeR) is designed to work with raw counts.</p>
<div class="sourceCode" id="cb134"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb134-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb134-1" tabindex="-1"></a><span class="fu">library</span>(vsn)</span>
<span id="cb134-2"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb134-2" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb134-3"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb134-3" tabindex="-1"></a></span>
<span id="cb134-4"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb134-4" tabindex="-1"></a>ntd <span class="ot">&lt;-</span> <span class="fu">normTransform</span>(dds) <span class="co"># log2(n + 1)</span></span>
<span id="cb134-5"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb134-5" tabindex="-1"></a></span>
<span id="cb134-6"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb134-6" tabindex="-1"></a>gg1 <span class="ot">&lt;-</span> <span class="fu">meanSdPlot</span>(<span class="fu">counts</span>(dds), <span class="at">plot=</span><span class="cn">FALSE</span>)[[<span class="st">&quot;gg&quot;</span>]] <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">&quot;Raw counts&quot;</span>)</span>
<span id="cb134-7"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb134-7" tabindex="-1"></a>gg2 <span class="ot">&lt;-</span> <span class="fu">meanSdPlot</span>(<span class="fu">assay</span>(ntd), <span class="at">plot=</span><span class="cn">FALSE</span>)[[<span class="st">&quot;gg&quot;</span>]] <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">&quot;log2(n+1)&quot;</span>)</span>
<span id="cb134-8"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb134-8" tabindex="-1"></a>gg3 <span class="ot">&lt;-</span> <span class="fu">meanSdPlot</span>(<span class="fu">assay</span>(vsd), <span class="at">plot=</span><span class="cn">FALSE</span>)[[<span class="st">&quot;gg&quot;</span>]] <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">&quot;Variance stabilizing transformed&quot;</span>)</span>
<span id="cb134-9"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb134-9" tabindex="-1"></a></span>
<span id="cb134-10"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb134-10" tabindex="-1"></a><span class="co"># use patchwork to combine the three plots into one figure.</span></span>
<span id="cb134-11"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb134-11" tabindex="-1"></a>(gg1 <span class="sc">|</span> gg2 <span class="sc">|</span> gg3) <span class="sc">+</span> <span class="fu">plot_layout</span>(<span class="at">nrow =</span> <span class="dv">1</span>)</span></code></pre></div>
<p><img src="03_rnaseq_workflow_files/figure-html/mv_plots-1.png" width="1152" /></p>
</div>
</div>
<div id="volcano-plot" class="section level2 hasAnchor" number="4.5">
<h2><span class="header-section-number">4.5</span> <strong>Volcano plot</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#volcano-plot" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>First, run the code below which makes a new function to make a volcano plot.</p>
<div class="sourceCode" id="cb135"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb135-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb135-1" tabindex="-1"></a>make_volcano <span class="ot">&lt;-</span> <span class="cf">function</span>(df, pval_nm, <span class="at">pval_cutoff=</span><span class="fl">0.1</span>){</span>
<span id="cb135-2"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb135-2" tabindex="-1"></a>  <span class="co"># remove genes with NA for pvalue</span></span>
<span id="cb135-3"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb135-3" tabindex="-1"></a>  df <span class="ot">&lt;-</span> df[<span class="fu">which</span>(<span class="sc">!</span><span class="fu">is.na</span>(df[[pval_nm]])), ]</span>
<span id="cb135-4"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb135-4" tabindex="-1"></a>  </span>
<span id="cb135-5"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb135-5" tabindex="-1"></a>  <span class="co"># add gene names</span></span>
<span id="cb135-6"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb135-6" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">cbind</span>(df, <span class="fu">rowData</span>(dds)[<span class="fu">rownames</span>(df), <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>])</span>
<span id="cb135-7"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb135-7" tabindex="-1"></a>  </span>
<span id="cb135-8"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb135-8" tabindex="-1"></a>  top_genes <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span></span>
<span id="cb135-9"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb135-9" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">arrange</span>(<span class="fu">desc</span>(<span class="fu">abs</span>(df<span class="sc">$</span>log2FoldChange))) <span class="sc">%&gt;%</span></span>
<span id="cb135-10"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb135-10" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">&lt;=</span> <span class="dv">10</span>) <span class="sc">%&gt;%</span></span>
<span id="cb135-11"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb135-11" tabindex="-1"></a>    <span class="fu">rownames</span>()</span>
<span id="cb135-12"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb135-12" tabindex="-1"></a>  </span>
<span id="cb135-13"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb135-13" tabindex="-1"></a>  df<span class="sc">$</span>Sig <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(df<span class="sc">$</span>padj <span class="sc">&lt;=</span> pval_cutoff, <span class="st">&quot;Sig&quot;</span>, <span class="st">&quot;NS&quot;</span>)</span>
<span id="cb135-14"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb135-14" tabindex="-1"></a>  </span>
<span id="cb135-15"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb135-15" tabindex="-1"></a>  df[[pval_nm]] <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(df[[pval_nm]])</span>
<span id="cb135-16"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb135-16" tabindex="-1"></a>  </span>
<span id="cb135-17"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb135-17" tabindex="-1"></a>  </span>
<span id="cb135-18"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb135-18" tabindex="-1"></a>  <span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x=</span>.data[[<span class="st">&quot;log2FoldChange&quot;</span>]], <span class="at">y=</span>.data[[pval_nm]])) <span class="sc">+</span></span>
<span id="cb135-19"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb135-19" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color=</span>Sig), <span class="at">size=</span><span class="fl">0.6</span>) <span class="sc">+</span></span>
<span id="cb135-20"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb135-20" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;black&quot;</span>, <span class="st">&quot;salmon&quot;</span>)) <span class="sc">+</span></span>
<span id="cb135-21"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb135-21" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">ylab</span>(<span class="fu">str_glue</span>(<span class="st">&quot;-log10(&quot;</span>, pval_nm,<span class="st">&quot;)&quot;</span>)) <span class="sc">+</span></span>
<span id="cb135-22"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb135-22" tabindex="-1"></a>    <span class="fu">geom_text_repel</span>(<span class="at">data=</span>df[top_genes, ],</span>
<span id="cb135-23"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb135-23" tabindex="-1"></a>                    <span class="fu">aes</span>(<span class="at">label=</span>Uniq_syms), <span class="at">max.overlaps=</span><span class="cn">Inf</span>, <span class="at">min.segment.length =</span> <span class="dv">0</span>)</span>
<span id="cb135-24"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb135-24" tabindex="-1"></a>}</span></code></pre></div>
<p>Run the command below to use the above function to make a volcano plot. Functions can help reduce the amount of code we have to write if we have to run similar code multiple times; for example, if we had multiple groups in our experiment and wanted to make volcano plots for several different contrasts.</p>
<div class="sourceCode" id="cb136"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb136-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb136-1" tabindex="-1"></a><span class="fu">make_volcano</span>(<span class="fu">as.data.frame</span>(lfc_shrink),</span>
<span id="cb136-2"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb136-2" tabindex="-1"></a>             <span class="at">pval_nm=</span><span class="st">&quot;padj&quot;</span>, <span class="at">pval_cutoff=</span>fdr_cutoff)</span></code></pre></div>
<p><img src="03_rnaseq_workflow_files/figure-html/volcano-1.png" width="384" /></p>
</div>
<div id="make-a-new-summarizedexperiment-to-store-all-results" class="section level2 hasAnchor" number="4.6">
<h2><span class="header-section-number">4.6</span> <strong>Make a new SummarizedExperiment to store all results</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#make-a-new-summarizedexperiment-to-store-all-results" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb137"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb137-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb137-1" tabindex="-1"></a><span class="co"># recall that we removed some genes new little or no expression.</span></span>
<span id="cb137-2"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb137-2" tabindex="-1"></a>se2 <span class="ot">&lt;-</span> se[<span class="fu">rownames</span>(dds), <span class="fu">colnames</span>(dds)]</span>
<span id="cb137-3"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb137-3" tabindex="-1"></a></span>
<span id="cb137-4"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb137-4" tabindex="-1"></a><span class="co"># replace the old variance-stabilized transformed counts</span></span>
<span id="cb137-5"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb137-5" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">identical</span>(<span class="fu">rownames</span>(se2), <span class="fu">rownames</span>(vsd)))</span>
<span id="cb137-6"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb137-6" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">identical</span>(<span class="fu">colnames</span>(se2), <span class="fu">colnames</span>(vsd)))</span>
<span id="cb137-7"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb137-7" tabindex="-1"></a><span class="fu">assay</span>(se2, <span class="st">&quot;vst&quot;</span>) <span class="ot">&lt;-</span> <span class="fu">assay</span>(vsd)</span>
<span id="cb137-8"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb137-8" tabindex="-1"></a></span>
<span id="cb137-9"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb137-9" tabindex="-1"></a><span class="co"># Add DE results</span></span>
<span id="cb137-10"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb137-10" tabindex="-1"></a><span class="fu">rowData</span>(se2)<span class="sc">$</span>trt.v.untrt.pval <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(lfc_shrink)[<span class="fu">rownames</span>(se2), <span class="st">&quot;pvalue&quot;</span>]</span>
<span id="cb137-11"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb137-11" tabindex="-1"></a><span class="fu">rowData</span>(se2)<span class="sc">$</span>trt.v.untrt.padj <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(lfc_shrink)[<span class="fu">rownames</span>(se2), <span class="st">&quot;padj&quot;</span>]</span>
<span id="cb137-12"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb137-12" tabindex="-1"></a><span class="fu">rowData</span>(se2)<span class="sc">$</span>trt.v.untrt.lfc <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(lfc_shrink)[<span class="fu">rownames</span>(se2), <span class="st">&quot;log2FoldChange&quot;</span>]</span>
<span id="cb137-13"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb137-13" tabindex="-1"></a><span class="fu">rowData</span>(se2)</span></code></pre></div>
<pre><code>## DataFrame with 21485 rows and 7 columns
##                          Symbol              Uniq_syms      entrez
##                     &lt;character&gt;            &lt;character&gt; &lt;character&gt;
## ENSG00000227232.5            NA      ENSG00000227232.5          NA
## ENSG00000278267.1     MIR6859-2 MIR6859-2_ENSG000002..   102465909
## ENSG00000241860.7            NA      ENSG00000241860.7          NA
## ENSG00000279457.4            NA      ENSG00000279457.4          NA
## ENSG00000228463.10    RPL23AP21              RPL23AP21      728481
## ...                         ...                    ...         ...
## ENSG00000273748.1            NA      ENSG00000273748.1          NA
## ENSG00000271254.6  LOC102724250           LOC102724250   102724250
## ENSG00000278673.1            NA      ENSG00000278673.1          NA
## ENSG00000278817.1         DGCR6 DGCR6_ENSG0000027881..        8214
## ENSG00000277196.4         PRODH PRODH_ENSG0000027719..        5625
##                                 Gene_name trt.v.untrt.pval trt.v.untrt.padj
##                               &lt;character&gt;        &lt;numeric&gt;        &lt;numeric&gt;
## ENSG00000227232.5                      NA         0.699875          0.88721
## ENSG00000278267.1         microRNA 6859-2         0.941844               NA
## ENSG00000241860.7                      NA         0.488979               NA
## ENSG00000279457.4                      NA         0.650232          0.86135
## ENSG00000228463.10 ribosomal protein L2..         0.446495               NA
## ...                                   ...              ...              ...
## ENSG00000273748.1                      NA      6.98054e-01      0.886856022
## ENSG00000271254.6  neuroblastoma breakp..      9.71630e-05      0.001446724
## ENSG00000278673.1                      NA      9.47906e-01               NA
## ENSG00000278817.1  DiGeorge syndrome cr..      3.02400e-02               NA
## ENSG00000277196.4  proline dehydrogenas..      1.47224e-05      0.000285074
##                    trt.v.untrt.lfc
##                          &lt;numeric&gt;
## ENSG00000227232.5       0.03692037
## ENSG00000278267.1       0.00642251
## ENSG00000241860.7       0.06385483
## ENSG00000279457.4       0.04216935
## ENSG00000228463.10      0.07241761
## ...                            ...
## ENSG00000273748.1      -0.03748339
## ENSG00000271254.6       0.66240874
## ENSG00000278673.1      -0.00502522
## ENSG00000278817.1      -0.28581346
## ENSG00000277196.4      -2.92428962</code></pre>
</div>
<div id="heatmap-of-top-de-genes" class="section level2 hasAnchor" number="4.7">
<h2><span class="header-section-number">4.7</span> <strong>Heatmap of top DE genes</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#heatmap-of-top-de-genes" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb139"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb139-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb139-1" tabindex="-1"></a>make_ht <span class="ot">&lt;-</span> <span class="cf">function</span>(se, genes, <span class="at">assay_nm=</span><span class="st">&quot;vst&quot;</span>, <span class="at">col_annot =</span> <span class="st">&quot;group&quot;</span>, <span class="at">row_label =</span> <span class="st">&quot;Uniq_syms&quot;</span>){</span>
<span id="cb139-2"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb139-2" tabindex="-1"></a>  ht_se <span class="ot">&lt;-</span> se[genes, ]</span>
<span id="cb139-3"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb139-3" tabindex="-1"></a>  coldat <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(ht_se)[, col_annot, <span class="at">drop=</span><span class="cn">FALSE</span>])</span>
<span id="cb139-4"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb139-4" tabindex="-1"></a>  coldat_lvls <span class="ot">&lt;-</span> <span class="fu">unique</span>(coldat[, <span class="dv">1</span>])</span>
<span id="cb139-5"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb139-5" tabindex="-1"></a>  ht_col_colors <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="fu">setNames</span>(<span class="at">nm=</span>coldat_lvls, <span class="at">object =</span> viridis<span class="sc">::</span><span class="fu">viridis</span>(<span class="fu">length</span>(coldat_lvls))))</span>
<span id="cb139-6"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb139-6" tabindex="-1"></a>  <span class="fu">names</span>(ht_col_colors) <span class="ot">&lt;-</span> col_annot</span>
<span id="cb139-7"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb139-7" tabindex="-1"></a>  </span>
<span id="cb139-8"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb139-8" tabindex="-1"></a>  <span class="fu">Heatmap</span>(<span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(<span class="fu">assay</span>(ht_se, assay_nm)), <span class="at">scale=</span><span class="cn">FALSE</span>, <span class="at">center =</span> <span class="cn">TRUE</span>)), </span>
<span id="cb139-9"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb139-9" tabindex="-1"></a>          <span class="at">name =</span> <span class="st">&quot;Mean-centered</span><span class="sc">\n</span><span class="st">expression&quot;</span>, </span>
<span id="cb139-10"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb139-10" tabindex="-1"></a>          <span class="at">cluster_columns =</span> <span class="cn">FALSE</span>,</span>
<span id="cb139-11"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb139-11" tabindex="-1"></a>          <span class="at">row_labels=</span><span class="fu">rowData</span>(ht_se)[, row_label],</span>
<span id="cb139-12"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb139-12" tabindex="-1"></a>          <span class="at">show_column_names =</span> <span class="cn">TRUE</span>,</span>
<span id="cb139-13"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb139-13" tabindex="-1"></a>          <span class="at">top_annotation=</span><span class="fu">HeatmapAnnotation</span>(<span class="at">df=</span>coldat, </span>
<span id="cb139-14"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb139-14" tabindex="-1"></a>                                           <span class="at">col=</span>ht_col_colors),</span>
<span id="cb139-15"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb139-15" tabindex="-1"></a>          <span class="at">row_title =</span> <span class="fu">paste0</span>(<span class="fu">nrow</span>(ht_se), <span class="st">&quot; genes&quot;</span>)</span>
<span id="cb139-16"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb139-16" tabindex="-1"></a>  )</span>
<span id="cb139-17"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb139-17" tabindex="-1"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb140"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb140-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb140-1" tabindex="-1"></a><span class="co"># sort genes in the se object based on significance.</span></span>
<span id="cb140-2"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb140-2" tabindex="-1"></a>se2 <span class="ot">&lt;-</span> se2[<span class="fu">order</span>(<span class="fu">rowData</span>(se2)<span class="sc">$</span>trt.v.untrt.pval), ]</span>
<span id="cb140-3"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb140-3" tabindex="-1"></a>top_genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(se2)[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]</span>
<span id="cb140-4"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb140-4" tabindex="-1"></a></span>
<span id="cb140-5"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb140-5" tabindex="-1"></a><span class="fu">make_ht</span>(<span class="at">se=</span>se2, <span class="at">genes=</span>top_genes, <span class="at">assay_nm=</span><span class="st">&quot;vst&quot;</span>, <span class="at">col_annot =</span> <span class="st">&quot;group&quot;</span>)</span></code></pre></div>
<p><img src="03_rnaseq_workflow_files/figure-html/top_degs_ht-1.png" width="672" /></p>
</div>
<div id="look-for-specific-genes" class="section level2 hasAnchor" number="4.8">
<h2><span class="header-section-number">4.8</span> <strong>Look for specific genes</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#look-for-specific-genes" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We know certain genes should be differentially expressed based on the <a href="https://pubmed.ncbi.nlm.nih.gov/24926665/">paper</a> that this dataset came from. We can check that these genes were significantly DE in our analysis.</p>
<div class="sourceCode" id="cb141"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb141-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb141-1" tabindex="-1"></a>gois <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;DUSP1&quot;</span>, <span class="st">&quot;KLF15&quot;</span>, <span class="st">&quot;CRISPLD2&quot;</span>)</span>
<span id="cb141-2"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb141-2" tabindex="-1"></a></span>
<span id="cb141-3"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb141-3" tabindex="-1"></a><span class="co"># we need to convert these to ensembl IDs</span></span>
<span id="cb141-4"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb141-4" tabindex="-1"></a>gois <span class="ot">&lt;-</span> <span class="fu">rownames</span>(se2)[<span class="fu">match</span>(gois, <span class="fu">rowData</span>(se2)[, <span class="st">&quot;Symbol&quot;</span>])]</span>
<span id="cb141-5"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb141-5" tabindex="-1"></a></span>
<span id="cb141-6"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb141-6" tabindex="-1"></a><span class="fu">make_ht</span>(<span class="at">se=</span>se2, <span class="at">genes=</span>gois, <span class="at">assay_nm=</span><span class="st">&quot;vst&quot;</span>, <span class="at">col_annot =</span> <span class="st">&quot;group&quot;</span>)</span></code></pre></div>
<p><img src="03_rnaseq_workflow_files/figure-html/specific_genes-1.png" width="672" /></p>
</div>
<div id="summarizedexperiment-exercises" class="section level2 hasAnchor" number="4.9">
<h2><span class="header-section-number">4.9</span> <strong>SummarizedExperiment exercises</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#summarizedexperiment-exercises" class="anchor-section" aria-label="Anchor link to header"></a></h2>
</div>
<div id="your-turn" class="section level2 hasAnchor" number="4.10">
<h2><span class="header-section-number">4.10</span> <strong>Your turn</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#your-turn" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ol style="list-style-type: decimal">
<li>Copy ‘se2’ to make a new object. Filter for significance then rank by the absolute value of LFC, largest to smallest.</li>
<li>Reorder the samples in the new object so that ‘untrt’ samples are on the left in the heatmap.</li>
<li>Plot top DEGs heatmap based on LFC ranking.</li>
</ol>
<details>
<summary>
Click here to see an example solution
</summary>
<div class="sourceCode" id="cb142"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb142-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb142-1" tabindex="-1"></a>se3 <span class="ot">&lt;-</span> se2</span>
<span id="cb142-2"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb142-2" tabindex="-1"></a>se3 <span class="ot">&lt;-</span> se3[<span class="fu">which</span>(<span class="fu">rowData</span>(se3)<span class="sc">$</span>trt.v.untrt.padj <span class="sc">&lt;=</span> <span class="fl">0.05</span>), ] <span class="co"># note that some rows have NAs in this column, so we use which()</span></span>
<span id="cb142-3"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb142-3" tabindex="-1"></a>se3 <span class="ot">&lt;-</span> se3[<span class="fu">order</span>(<span class="fu">abs</span>(<span class="fu">rowData</span>(se3)<span class="sc">$</span>trt.v.untrt.lfc), <span class="at">decreasing =</span> <span class="cn">TRUE</span>), ]</span>
<span id="cb142-4"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb142-4" tabindex="-1"></a>se3 <span class="ot">&lt;-</span> se3[, <span class="fu">rev</span>(<span class="fu">order</span>(se3<span class="sc">$</span>group))]</span>
<span id="cb142-5"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb142-5" tabindex="-1"></a></span>
<span id="cb142-6"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb142-6" tabindex="-1"></a>top_genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(se3)[<span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>]</span>
<span id="cb142-7"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb142-7" tabindex="-1"></a></span>
<span id="cb142-8"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb142-8" tabindex="-1"></a><span class="fu">make_ht</span>(<span class="at">se=</span>se3, <span class="at">genes=</span>top_genes, <span class="at">assay_nm=</span><span class="st">&quot;vst&quot;</span>, <span class="at">col_annot =</span> <span class="st">&quot;group&quot;</span>)</span></code></pre></div>
</details>
<p><br></p>
</div>
<div id="venn-diagrams" class="section level2 hasAnchor" number="4.11">
<h2><span class="header-section-number">4.11</span> <strong>Venn diagrams</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#venn-diagrams" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>One option is <a href="https://cran.r-project.org/web/packages/ggVennDiagram/readme/README.html">ggVennDiagram</a>.</p>
</div>
<div id="upset-plots" class="section level2 hasAnchor" number="4.12">
<h2><span class="header-section-number">4.12</span> <strong>Upset plots</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#upset-plots" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We recommend using the <a href="https://jokergoo.github.io/ComplexHeatmap-reference/book/upset-plot.html">ComplexHeatmap package</a>.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="mini-project.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="pathway-and-enrichment-analysis.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="assets/gitbook-2.6.7/js/app.min.js"></script>
<script src="assets/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="assets/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="assets/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="assets/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="assets/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="assets/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="assets/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": true,
    "facebook": false,
    "twitter": false,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": false
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": null,
    "text": null
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": null,
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  },
  "toolbar": {
    "position": "static"
  }
});
});
</script>

</body>

</html>
