<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Part 3 Mini Project | Introductory Bioinformatics for VAI Users</title>
  <meta name="description" content="Bioinformatics tutorials with hands-on exercises and examples." />
  <meta name="generator" content="bookdown 0.42 and GitBook 2.6.7" />

  <meta property="og:title" content="Part 3 Mini Project | Introductory Bioinformatics for VAI Users" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Bioinformatics tutorials with hands-on exercises and examples." />
  <meta name="github-repo" content="vari-bbc/Intro_to_Linux_for_HPC" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Part 3 Mini Project | Introductory Bioinformatics for VAI Users" />
  
  <meta name="twitter:description" content="Bioinformatics tutorials with hands-on exercises and examples." />
  

<meta name="author" content="The Bioinformatics and Biostatistics Core (BBC)" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="basic-command-line-utilities-on-the-hpc.html"/>
<link rel="next" href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"/>
<script src="assets/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="assets/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="assets/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="assets/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="assets/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="assets/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="assets/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="assets/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="assets/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="assets/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="assets/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="assets/kePrint-0.0.1/kePrint.js"></script>
<link href="assets/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> <strong>Introduction</strong></a></li>
<li class="chapter" data-level="2" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html"><i class="fa fa-check"></i><b>2</b> <strong>Basic Command Line Utilities on the HPC</strong></a>
<ul>
<li class="chapter" data-level="2.1" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#access-hpc"><i class="fa fa-check"></i><b>2.1</b> <strong>Access HPC</strong></a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#ondemand-instructions"><i class="fa fa-check"></i><b>2.1.1</b> <strong>OnDemand Instructions</strong></a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#file-navigation"><i class="fa fa-check"></i><b>2.2</b> <strong>File navigation</strong></a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#navigate-to-a-directory-folder"><i class="fa fa-check"></i><b>2.2.1</b> <strong>Navigate to a directory (folder)</strong></a></li>
<li class="chapter" data-level="2.2.2" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#list-content-in-a-directory"><i class="fa fa-check"></i><b>2.2.2</b> <strong>List content in a directory</strong></a></li>
<li class="chapter" data-level="2.2.3" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#navigate-back-to-home-directory"><i class="fa fa-check"></i><b>2.2.3</b> <strong>Navigate back to home directory</strong></a></li>
<li class="chapter" data-level="2.2.4" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#create-a-directory"><i class="fa fa-check"></i><b>2.2.4</b> <strong>Create a directory</strong></a></li>
<li class="chapter" data-level="2.2.5" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#copy-a-file"><i class="fa fa-check"></i><b>2.2.5</b> <strong>Copy a file</strong></a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#view-and-manipulate-files"><i class="fa fa-check"></i><b>2.3</b> <strong>View and manipulate files</strong></a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#display-content"><i class="fa fa-check"></i><b>2.3.1</b> <strong>Display content</strong></a></li>
<li class="chapter" data-level="2.3.2" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#pattern-search"><i class="fa fa-check"></i><b>2.3.2</b> <strong>Pattern search</strong></a></li>
<li class="chapter" data-level="2.3.3" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#display-the-number-of-lines-words-and-characters"><i class="fa fa-check"></i><b>2.3.3</b> <strong>Display the number of lines, words, and characters</strong></a></li>
<li class="chapter" data-level="2.3.4" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#pipe---redirection"><i class="fa fa-check"></i><b>2.3.4</b> <strong>Pipe - redirection</strong></a></li>
<li class="chapter" data-level="2.3.5" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#output-redirection"><i class="fa fa-check"></i><b>2.3.5</b> <strong>Output redirection</strong></a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#exercise"><i class="fa fa-check"></i><b>2.4</b> <strong>Exercise</strong></a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#step-1-copy-the-files"><i class="fa fa-check"></i><b>2.4.1</b> <strong>Step 1: Copy the files</strong></a></li>
<li class="chapter" data-level="2.4.2" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#step-2-combine-the-files"><i class="fa fa-check"></i><b>2.4.2</b> <strong>Step 2: Combine the files</strong></a></li>
<li class="chapter" data-level="2.4.3" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#step-3-count-the-number-of-reads"><i class="fa fa-check"></i><b>2.4.3</b> <strong>Step 3: Count the number of reads</strong></a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#summary"><i class="fa fa-check"></i><b>2.5</b> <strong>Summary</strong></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="mini-project.html"><a href="mini-project.html"><i class="fa fa-check"></i><b>3</b> <strong>Mini Project</strong></a>
<ul>
<li class="chapter" data-level="3.1" data-path="mini-project.html"><a href="mini-project.html#start-an-interactive-job"><i class="fa fa-check"></i><b>3.1</b> <strong>Start an Interactive Job</strong></a></li>
<li class="chapter" data-level="3.2" data-path="mini-project.html"><a href="mini-project.html#create-directory"><i class="fa fa-check"></i><b>3.2</b> <strong>Create Directory</strong></a></li>
<li class="chapter" data-level="3.3" data-path="mini-project.html"><a href="mini-project.html#copy-fastq-files-to-fastqs-subdirectory"><i class="fa fa-check"></i><b>3.3</b> <strong>Copy Fastq Files To <code>fastqs</code> subdirectory</strong></a></li>
<li class="chapter" data-level="3.4" data-path="mini-project.html"><a href="mini-project.html#use-basic-linux-commands-to-explore-the-fastq-files"><i class="fa fa-check"></i><b>3.4</b> <strong>Use basic Linux commands to explore the fastq files</strong></a></li>
<li class="chapter" data-level="3.5" data-path="mini-project.html"><a href="mini-project.html#working-with-environment-modules"><i class="fa fa-check"></i><b>3.5</b> <strong>Working With Environment Modules</strong></a></li>
<li class="chapter" data-level="3.6" data-path="mini-project.html"><a href="mini-project.html#run-fastqc"><i class="fa fa-check"></i><b>3.6</b> <strong>Run FastQC</strong></a></li>
<li class="chapter" data-level="3.7" data-path="mini-project.html"><a href="mini-project.html#set-up-a-job-script-to-run-salmon-to-align-the-reads"><i class="fa fa-check"></i><b>3.7</b> <strong>Set up a job script to run Salmon to align the reads</strong></a></li>
<li class="chapter" data-level="3.8" data-path="mini-project.html"><a href="mini-project.html#submit-a-job"><i class="fa fa-check"></i><b>3.8</b> <strong>Submit a Job</strong></a></li>
<li class="chapter" data-level="3.9" data-path="mini-project.html"><a href="mini-project.html#examine-the-standard-output-stdout-and-standard-error-stderr-log-files"><i class="fa fa-check"></i><b>3.9</b> <strong>Examine the Standard Output (stdout) and Standard Error (stderr) log files</strong></a></li>
<li class="chapter" data-level="3.10" data-path="mini-project.html"><a href="mini-project.html#use-grep-to-find-the-tpms-for-a-specific-gene"><i class="fa fa-check"></i><b>3.10</b> <strong>Use grep to find the TPMs for a specific gene</strong></a></li>
<li class="chapter" data-level="3.11" data-path="mini-project.html"><a href="mini-project.html#bonus-use-an-interactive-job-to-run-multiqc-on-the-salmon-and-fastqc-output"><i class="fa fa-check"></i><b>3.11</b> <strong>BONUS: Use an interactive job to run multiQC on the Salmon and FastQC output</strong></a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><i class="fa fa-check"></i><b>4</b> <strong>Basic RNA-Seq Analysis – From fastq files to DE genes</strong></a>
<ul>
<li class="chapter" data-level="4.1" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#run-the-bbc-pipeline-to-align-the-reads"><i class="fa fa-check"></i><b>4.1</b> <strong>Run the BBC pipeline to align the reads</strong></a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#the-workflow-as-a-graph"><i class="fa fa-check"></i><b>4.1.1</b> <strong>The workflow as a graph</strong></a></li>
<li class="chapter" data-level="4.1.2" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#clone-the-github-repo"><i class="fa fa-check"></i><b>4.1.2</b> <strong>Clone the Github repo</strong></a></li>
<li class="chapter" data-level="4.1.3" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#add-your-fastq-files-to-raw_data"><i class="fa fa-check"></i><b>4.1.3</b> <strong>Add your fastq files to <code>raw_data/</code></strong></a></li>
<li class="chapter" data-level="4.1.4" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#fill-out-the-samplesheet"><i class="fa fa-check"></i><b>4.1.4</b> <strong>Fill out the samplesheet</strong></a></li>
<li class="chapter" data-level="4.1.5" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#fill-out-the-config-file"><i class="fa fa-check"></i><b>4.1.5</b> <strong>Fill out the config file</strong></a></li>
<li class="chapter" data-level="4.1.6" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#submit-the-main-snakemake-job"><i class="fa fa-check"></i><b>4.1.6</b> <strong>Submit the main Snakemake job</strong></a></li>
<li class="chapter" data-level="4.1.7" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#bbc-maintained-reference-files"><i class="fa fa-check"></i><b>4.1.7</b> <strong>BBC-maintained reference files</strong></a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#symlink-pre-run-results-to-your-directory"><i class="fa fa-check"></i><b>4.2</b> <strong>Symlink pre-run results to your directory</strong></a></li>
<li class="chapter" data-level="4.3" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#quality-control"><i class="fa fa-check"></i><b>4.3</b> <strong>Quality control</strong></a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#multiqc-report"><i class="fa fa-check"></i><b>4.3.1</b> <strong>multiQC report</strong></a></li>
<li class="chapter" data-level="4.3.2" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#other-considerations"><i class="fa fa-check"></i><b>4.3.2</b> <strong>Other considerations</strong></a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#de-analysis-using-r-and-deseq2"><i class="fa fa-check"></i><b>4.4</b> <strong>DE analysis using R and DESeq2</strong></a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#create-an-r-project"><i class="fa fa-check"></i><b>4.4.1</b> <strong>Create an R project</strong></a></li>
<li class="chapter" data-level="4.4.2" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#exploring-the-data-using-isee"><i class="fa fa-check"></i><b>4.4.2</b> <strong>Exploring the data using iSEE</strong></a></li>
<li class="chapter" data-level="4.4.3" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#make-a-separate-folder-for-de-analysis-and-a-script-file-to-store-our-commands"><i class="fa fa-check"></i><b>4.4.3</b> <strong>Make a separate folder for DE analysis and a script file to store our commands</strong></a></li>
<li class="chapter" data-level="4.4.4" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#make-an-output-directory"><i class="fa fa-check"></i><b>4.4.4</b> <strong>Make an output directory</strong></a></li>
<li class="chapter" data-level="4.4.5" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#install-packages"><i class="fa fa-check"></i><b>4.4.5</b> <strong>Install packages</strong></a></li>
<li class="chapter" data-level="4.4.6" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#load-packages"><i class="fa fa-check"></i><b>4.4.6</b> <strong>Load packages</strong></a></li>
<li class="chapter" data-level="4.4.7" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#set-up-your-deseq-object"><i class="fa fa-check"></i><b>4.4.7</b> <strong>Set up your DESeq object</strong></a></li>
<li class="chapter" data-level="4.4.8" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#what-is-a-summarizedexperiment-object"><i class="fa fa-check"></i><b>4.4.8</b> <strong>What is a SummarizedExperiment object?</strong></a></li>
<li class="chapter" data-level="4.4.9" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#remove-genes-with-lowno-expression"><i class="fa fa-check"></i><b>4.4.9</b> <strong>Remove genes with low/no expression</strong></a></li>
<li class="chapter" data-level="4.4.10" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#different-normalization-approaches-for-different-biases"><i class="fa fa-check"></i><b>4.4.10</b> <strong>Different normalization approaches for different biases</strong></a></li>
<li class="chapter" data-level="4.4.11" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#run-the-de-workflow"><i class="fa fa-check"></i><b>4.4.11</b> <strong>Run the DE workflow</strong></a></li>
<li class="chapter" data-level="4.4.12" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#summarize-de-results"><i class="fa fa-check"></i><b>4.4.12</b> <strong>Summarize DE results</strong></a></li>
<li class="chapter" data-level="4.4.13" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#p-value-distribution"><i class="fa fa-check"></i><b>4.4.13</b> <strong>P value distribution</strong></a></li>
<li class="chapter" data-level="4.4.14" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#shrink-log-fold-changes-for-lowly-expressed-genes"><i class="fa fa-check"></i><b>4.4.14</b> <strong>Shrink log fold changes for lowly expressed genes</strong></a></li>
<li class="chapter" data-level="4.4.15" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#output-de-results"><i class="fa fa-check"></i><b>4.4.15</b> <strong>Output DE results</strong></a></li>
<li class="chapter" data-level="4.4.16" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#output-tables-with-log2-normalized-counts"><i class="fa fa-check"></i><b>4.4.16</b> <strong>Output tables with log2 normalized counts</strong></a></li>
<li class="chapter" data-level="4.4.17" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#why-should-we-transform-counts-for-visualizations-or-clustering"><i class="fa fa-check"></i><b>4.4.17</b> <strong>Why should we transform counts for visualizations or clustering?</strong></a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#volcano-plot"><i class="fa fa-check"></i><b>4.5</b> <strong>Volcano plot</strong></a></li>
<li class="chapter" data-level="4.6" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#make-a-new-summarizedexperiment-to-store-all-results"><i class="fa fa-check"></i><b>4.6</b> <strong>Make a new SummarizedExperiment to store all results</strong></a></li>
<li class="chapter" data-level="4.7" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#heatmap-of-top-de-genes"><i class="fa fa-check"></i><b>4.7</b> <strong>Heatmap of top DE genes</strong></a></li>
<li class="chapter" data-level="4.8" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#look-for-specific-genes"><i class="fa fa-check"></i><b>4.8</b> <strong>Look for specific genes</strong></a></li>
<li class="chapter" data-level="4.9" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#summarizedexperiment-exercises"><i class="fa fa-check"></i><b>4.9</b> <strong>SummarizedExperiment exercises</strong></a></li>
<li class="chapter" data-level="4.10" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#your-turn"><i class="fa fa-check"></i><b>4.10</b> <strong>Your turn</strong></a></li>
<li class="chapter" data-level="4.11" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#venn-diagrams"><i class="fa fa-check"></i><b>4.11</b> <strong>Venn diagrams</strong></a></li>
<li class="chapter" data-level="4.12" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#upset-plots"><i class="fa fa-check"></i><b>4.12</b> <strong>Upset plots</strong></a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="pathway-and-enrichment-analysis.html"><a href="pathway-and-enrichment-analysis.html"><i class="fa fa-check"></i><b>5</b> <strong>Pathway and enrichment analysis</strong></a>
<ul>
<li class="chapter" data-level="5.1" data-path="pathway-and-enrichment-analysis.html"><a href="pathway-and-enrichment-analysis.html#downstream-of-de-gene-analysis"><i class="fa fa-check"></i><b>5.1</b> <strong>Downstream of DE gene analysis</strong></a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="pathway-and-enrichment-analysis.html"><a href="pathway-and-enrichment-analysis.html#load-packages-1"><i class="fa fa-check"></i><b>5.1.1</b> <strong>load packages</strong></a></li>
<li class="chapter" data-level="5.1.2" data-path="pathway-and-enrichment-analysis.html"><a href="pathway-and-enrichment-analysis.html#import-previous-de-results"><i class="fa fa-check"></i><b>5.1.2</b> <strong>Import previous DE results</strong></a></li>
<li class="chapter" data-level="5.1.3" data-path="pathway-and-enrichment-analysis.html"><a href="pathway-and-enrichment-analysis.html#how-many-de-genes"><i class="fa fa-check"></i><b>5.1.3</b> <strong>How many DE genes</strong></a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="pathway-and-enrichment-analysis.html"><a href="pathway-and-enrichment-analysis.html#why-enrichment-and-pathway-analysis"><i class="fa fa-check"></i><b>5.2</b> <strong>Why enrichment and pathway analysis?</strong></a></li>
<li class="chapter" data-level="5.3" data-path="pathway-and-enrichment-analysis.html"><a href="pathway-and-enrichment-analysis.html#ora"><i class="fa fa-check"></i><b>5.3</b> <strong>ORA</strong></a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="pathway-and-enrichment-analysis.html"><a href="pathway-and-enrichment-analysis.html#plot-ora-results"><i class="fa fa-check"></i><b>5.3.1</b> <strong>plot ORA results</strong></a></li>
<li class="chapter" data-level="5.3.2" data-path="pathway-and-enrichment-analysis.html"><a href="pathway-and-enrichment-analysis.html#save-ora-results-to-a-tsv-file"><i class="fa fa-check"></i><b>5.3.2</b> <strong>save ORA results to a tsv file</strong></a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="pathway-and-enrichment-analysis.html"><a href="pathway-and-enrichment-analysis.html#gsea"><i class="fa fa-check"></i><b>5.4</b> <strong>GSEA</strong></a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="pathway-and-enrichment-analysis.html"><a href="pathway-and-enrichment-analysis.html#save-gsea-results"><i class="fa fa-check"></i><b>5.4.1</b> <strong>save gsea results</strong></a></li>
<li class="chapter" data-level="5.4.2" data-path="pathway-and-enrichment-analysis.html"><a href="pathway-and-enrichment-analysis.html#plot-gsea-results"><i class="fa fa-check"></i><b>5.4.2</b> <strong>plot GSEA results</strong></a></li>
<li class="chapter" data-level="5.4.3" data-path="pathway-and-enrichment-analysis.html"><a href="pathway-and-enrichment-analysis.html#plot-kegg-map"><i class="fa fa-check"></i><b>5.4.3</b> <strong>plot KEGG map</strong></a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="pathway-and-enrichment-analysis.html"><a href="pathway-and-enrichment-analysis.html#other-r-packages-and-tools"><i class="fa fa-check"></i><b>5.5</b> <strong>Other R packages and tools</strong></a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="appendices.html"><a href="appendices.html"><i class="fa fa-check"></i><b>6</b> <strong>Appendices</strong></a>
<ul>
<li class="chapter" data-level="6.1" data-path="appendices.html"><a href="appendices.html#other-workshop-files"><i class="fa fa-check"></i><b>6.1</b> <strong>Other workshop files</strong></a></li>
<li class="chapter" data-level="6.2" data-path="appendices.html"><a href="appendices.html#local-access-of-files-on-the-hpc"><i class="fa fa-check"></i><b>6.2</b> <strong>Local access of files on the HPC</strong></a></li>
<li class="chapter" data-level="6.3" data-path="appendices.html"><a href="appendices.html#bash-cheatsheet"><i class="fa fa-check"></i><b>6.3</b> <strong>Bash cheatsheet</strong></a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Introductory Bioinformatics for VAI Users</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="mini-project" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">Part 3</span> <strong>Mini Project</strong><a href="mini-project.html#mini-project" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Here, we will combine some of the commands we have learned today to perform a toy bioinformatic analysis. Two human paired-end samples (four total fastq files) have already been downloaded from <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE52778">GSE52778</a> and are in the directory <code>/varidata/researchtemp/hpctmp/BBC_workshop_Oct2024_II/fastqs</code>. To save time for the workshop, each file has been randomly subsetted to just a small fraction of the total reads. Here each user will:</p>
<ol style="list-style-type: decimal">
<li>Copy these fastq files their own private project directory.</li>
<li>Use basic Linux commands to learn some characteristics of these files.</li>
<li>Use Salmon to align (pseudo-align) these reads to the hg38 reference transcriptome and get transcripts-per-million (TPMs) for each annotated gene.</li>
</ol>
<div id="start-an-interactive-job" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> <strong>Start an Interactive Job</strong><a href="mini-project.html#start-an-interactive-job" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>While simple file manipulations can be done on the submit node, computationally intensive operations should be performed using preallocated computational resources. An <strong>interactive job</strong> allows us to preallocate resources while still being able to run commands line by line.</p>
<p>We will start an interactive job, requesting one CPU core and 1 hour and 30 minutes of walltime.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb23-1"><a href="mini-project.html#cb23-1" tabindex="-1"></a><span class="ex">srun</span> <span class="at">-p</span> short <span class="at">--nodes</span><span class="op">=</span>1 <span class="at">--ntasks-per-node</span><span class="op">=</span>1 <span class="at">--time</span><span class="op">=</span>01:30:00 <span class="at">--pty</span> bash </span></code></pre></div>
<p>After a job has been submitted, users can check on the status (e.g. how long it has been running) of it using the following.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb24-1"><a href="mini-project.html#cb24-1" tabindex="-1"></a><span class="ex">squeue</span> <span class="at">--me</span></span></code></pre></div>
</div>
<div id="create-directory" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> <strong>Create Directory</strong><a href="mini-project.html#create-directory" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Typically, one would work in their specific lab’s folder on the HPC. For this workshop, we will work in a common directory so that the instructors and check on your progress.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb25-1"><a href="mini-project.html#cb25-1" tabindex="-1"></a><span class="bu">cd</span> /varidata/researchtemp/hpctmp/BBC_workshop_Oct2024_II/</span></code></pre></div>
<p>Create a directory based on your username to separate your work from other users’.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb26-1"><a href="mini-project.html#cb26-1" tabindex="-1"></a><span class="fu">mkdir</span> <span class="op">&lt;</span>username<span class="op">&gt;</span></span></code></pre></div>
<p>Navigate to the username directory</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb27-1"><a href="mini-project.html#cb27-1" tabindex="-1"></a><span class="bu">cd</span> <span class="op">&lt;</span>username<span class="op">&gt;</span></span></code></pre></div>
<p>Create a <code>fastqs</code> directory inside the username directory. It is good practice to keep your raw data separate from your analyses and never make changes to them directly.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb28-1"><a href="mini-project.html#cb28-1" tabindex="-1"></a><span class="fu">mkdir</span> fastqs</span></code></pre></div>
<p>Use <code>ls</code> to confirm the <code>fastqs</code> directory was created.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb29-1"><a href="mini-project.html#cb29-1" tabindex="-1"></a><span class="fu">ls</span></span></code></pre></div>
</div>
<div id="copy-fastq-files-to-fastqs-subdirectory" class="section level2 hasAnchor" number="3.3">
<h2><span class="header-section-number">3.3</span> <strong>Copy Fastq Files To <code>fastqs</code> subdirectory</strong><a href="mini-project.html#copy-fastq-files-to-fastqs-subdirectory" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Verify the current directory is in the username directory.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb30-1"><a href="mini-project.html#cb30-1" tabindex="-1"></a><span class="bu">pwd</span></span></code></pre></div>
<p>Copy the 4 fastq.gz files and <code>md5sum.txt</code>, which you will use to check the validty of the files.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb31-1"><a href="mini-project.html#cb31-1" tabindex="-1"></a><span class="fu">cp</span> /varidata/researchtemp/hpctmp/BBC_workshop_Oct2024_II/fastqs/<span class="pp">*</span>fastq.gz ./fastqs/</span>
<span id="cb31-2"><a href="mini-project.html#cb31-2" tabindex="-1"></a></span>
<span id="cb31-3"><a href="mini-project.html#cb31-3" tabindex="-1"></a><span class="fu">cp</span> /varidata/researchtemp/hpctmp/BBC_workshop_Oct2024_II/fastqs/md5sum.txt ./fastqs/</span></code></pre></div>
<p>Verify that the copied files are valid. The following code should return with an <strong>OK</strong> for each file.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb32-1"><a href="mini-project.html#cb32-1" tabindex="-1"></a><span class="bu">cd</span> fastqs</span>
<span id="cb32-2"><a href="mini-project.html#cb32-2" tabindex="-1"></a></span>
<span id="cb32-3"><a href="mini-project.html#cb32-3" tabindex="-1"></a><span class="fu">md5sum</span> <span class="at">-c</span> md5sum.txt</span></code></pre></div>
<pre><code>## SRR1039520_1.fastq.gz: OK
## SRR1039520_2.fastq.gz: OK
## SRR1039521_1.fastq.gz: OK
## SRR1039521_2.fastq.gz: OK</code></pre>
<p>Go back to the username directory.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb34-1"><a href="mini-project.html#cb34-1" tabindex="-1"></a><span class="bu">cd</span> ..</span></code></pre></div>
</div>
<div id="use-basic-linux-commands-to-explore-the-fastq-files" class="section level2 hasAnchor" number="3.4">
<h2><span class="header-section-number">3.4</span> <strong>Use basic Linux commands to explore the fastq files</strong><a href="mini-project.html#use-basic-linux-commands-to-explore-the-fastq-files" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Notice all of the sequence data files end with a <code>.gz</code> extension, indicating they are gzip compressed. To view such files, one must use <code>zcat</code> instead of <code>cat</code> so that the files are decompressed into human-readable format.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb35-1"><a href="mini-project.html#cb35-1" tabindex="-1"></a><span class="fu">zcat</span> fastqs/SRR1039520_1.fastq.gz <span class="kw">|</span> <span class="fu">head</span></span></code></pre></div>
<pre><code>## @SRR1039520.19151550 19151550 length=63
## CCAGGACATCAAGAAGCCAGCTGAAGATGAGTGGGGTAAAACCCCAGACGCCATGAAAGCTGC
## +
## HJJJJJJJJJJJJJJJIJJJJJJJJIJJJJJGGJJJFHIJIIJJIGHHFFFDDDDDCDDDCDD
## @SRR1039520.3404519 3404519 length=63
## TGAGACATGGTTATAGATAAGAGAGTACAAAATGACTCTTTTTCCTGTCAATTGAAATTTAAA
## +
## HIGDFBGIIJBHGIIEHIIJIJIIIEGGIIIHGIJJIJJJIIJGIIIIGIEHIGDCHHIICG7
## @SRR1039520.16253787 16253787 length=63
## CAGGAGACCAAAGACACTGCAATTTGTGTGTTTTCTACAGGGTGCTTTAGATGACGTCTCATT</code></pre>
<div class="sourceCode" id="cb37"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb37-1"><a href="mini-project.html#cb37-1" tabindex="-1"></a><span class="fu">zcat</span> fastqs/SRR1039520_2.fastq.gz <span class="kw">|</span> <span class="fu">head</span></span></code></pre></div>
<pre><code>## @SRR1039520.19151550 19151550 length=63
## GAGATGGGGGTCCGTGCGGGCAGAACCCAGGGCATGAAGATCCAAAAGGGCCTGGTTCAGCTT
## +
## HJJJJJJJJJHHHIGIJIJJJJJJJHHHHFFFDDEEDDDDDDDDDDDDDDDDDDDDDDDDDDD
## @SRR1039520.3404519 3404519 length=63
## TTTGACCCTAGTATTGGCAATAGCCCTTTGCTATTTATATAATTAAAACTTTTCTTTAAATTT
## +
## HIJIJJJJIIEHGIGIIIDGIJJJJIIJJJIJJJIJJJIIIJJJJIJJGCHIJJJJJIJIJII
## @SRR1039520.16253787 16253787 length=63
## TACAGTTTGCAAAAGATGTCCAGATGGGTTCTTCTCAAATGAGACGTCATCTAAAGCACCCTG</code></pre>
<p>Notice that <code>SRR1039520_1.fastq.gz</code> and <code>SRR1039520_2.fastq.gz</code> files have the same read IDs. Valid paired fastq files always have matching read IDs throughout the entire file.</p>
<p>Next, we can figure out how many reads are in a fastq file using <code>wc -l</code>. Recall that each read is represented by 4 lines in a fastq file, so we need to divide the results of <code>wc -l</code> by 4 to get the number of reads.</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb39-1"><a href="mini-project.html#cb39-1" tabindex="-1"></a><span class="fu">zcat</span> fastqs/SRR1039520_1.fastq.gz <span class="kw">|</span> <span class="fu">wc</span> <span class="at">-l</span></span></code></pre></div>
<pre><code>## 2000000</code></pre>
<p>Finally, we can use <code>wc -m</code> to figure out the read length in a fastq file. Below, we use a combination of <code>head -n2</code> and <code>tail -n1</code> to get the second line of the first read. You may notice that the results of <code>wc -m</code> will be 1 higher than the actual read length. This is because the tool counts the newline character also.</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb41-1"><a href="mini-project.html#cb41-1" tabindex="-1"></a><span class="fu">zcat</span> fastqs/SRR1039520_1.fastq.gz <span class="kw">|</span> <span class="fu">head</span> <span class="at">-n2</span> <span class="kw">|</span> <span class="fu">tail</span> <span class="at">-n1</span> <span class="kw">|</span> <span class="fu">wc</span> <span class="at">-m</span></span></code></pre></div>
<pre><code>## 64</code></pre>
</div>
<div id="working-with-environment-modules" class="section level2 hasAnchor" number="3.5">
<h2><span class="header-section-number">3.5</span> <strong>Working With Environment Modules</strong><a href="mini-project.html#working-with-environment-modules" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Type <code>module av bbc2</code> (modules beginning with just <code>bbc</code> are from the old HPC) to see all the software installed by the BBC. There are a lot of modules installed; to parse through these more easily, we can <strong>pipe</strong> the results of <code>module av</code> into <code>grep</code> to search for modules with specific keywords. There is a trick, though. The results of <code>module av</code> go to standard error, so we need to redirect standard error to standard out using <code>2&gt;&amp;1</code> before the pipe.</p>
<p>This command will output any modules containing the <code>fastqc</code> keyword.</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb43-1"><a href="mini-project.html#cb43-1" tabindex="-1"></a><span class="ex">module</span> av <span class="at">-w</span> 1 bbc2 <span class="dv">2</span><span class="op">&gt;&amp;</span><span class="dv">1</span> <span class="kw">|</span> <span class="fu">grep</span> <span class="st">&#39;fastqc&#39;</span></span></code></pre></div>
<pre><code>## bbc2/fastqc/fastqc-0.12.1</code></pre>
</div>
<div id="run-fastqc" class="section level2 hasAnchor" number="3.6">
<h2><span class="header-section-number">3.6</span> <strong>Run FastQC</strong><a href="mini-project.html#run-fastqc" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Create a <code>fastqc</code> directory inside of your username directory and run FastQC.</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb45-1"><a href="mini-project.html#cb45-1" tabindex="-1"></a><span class="ex">module</span> load bbc2/fastqc/fastqc-0.12.1</span>
<span id="cb45-2"><a href="mini-project.html#cb45-2" tabindex="-1"></a></span>
<span id="cb45-3"><a href="mini-project.html#cb45-3" tabindex="-1"></a><span class="fu">mkdir</span> fastqc</span>
<span id="cb45-4"><a href="mini-project.html#cb45-4" tabindex="-1"></a></span>
<span id="cb45-5"><a href="mini-project.html#cb45-5" tabindex="-1"></a><span class="co"># Run FastQC. You can also run `fastqc -h` to see what different options do.</span></span>
<span id="cb45-6"><a href="mini-project.html#cb45-6" tabindex="-1"></a><span class="ex">fastqc</span> <span class="at">-o</span> fastqc fastqs/<span class="pp">*</span>fastq.gz</span></code></pre></div>
<pre><code>## Loading bbc2/fastqc/fastqc-0.12.1
##   Loading requirement: VARI/java/1.8.0_202
## application/gzip
## application/gzip
## Started analysis of SRR1039520_1.fastq.gz
## application/gzip
## application/gzip
## Approx 5% complete for SRR1039520_1.fastq.gz
## Approx 10% complete for SRR1039520_1.fastq.gz
## Approx 15% complete for SRR1039520_1.fastq.gz
## Approx 20% complete for SRR1039520_1.fastq.gz
## Approx 25% complete for SRR1039520_1.fastq.gz
## Approx 30% complete for SRR1039520_1.fastq.gz
## Approx 35% complete for SRR1039520_1.fastq.gz
## Approx 40% complete for SRR1039520_1.fastq.gz
## Approx 45% complete for SRR1039520_1.fastq.gz
## Approx 50% complete for SRR1039520_1.fastq.gz
## Approx 55% complete for SRR1039520_1.fastq.gz
## Approx 60% complete for SRR1039520_1.fastq.gz
## Approx 65% complete for SRR1039520_1.fastq.gz
## Approx 70% complete for SRR1039520_1.fastq.gz
## Approx 75% complete for SRR1039520_1.fastq.gz
## Approx 80% complete for SRR1039520_1.fastq.gz
## Approx 85% complete for SRR1039520_1.fastq.gz
## Approx 90% complete for SRR1039520_1.fastq.gz
## Approx 95% complete for SRR1039520_1.fastq.gz
## Approx 100% complete for SRR1039520_1.fastq.gz
## Analysis complete for SRR1039520_1.fastq.gz
## Started analysis of SRR1039520_2.fastq.gz
## Approx 5% complete for SRR1039520_2.fastq.gz
## Approx 10% complete for SRR1039520_2.fastq.gz
## Approx 15% complete for SRR1039520_2.fastq.gz
## Approx 20% complete for SRR1039520_2.fastq.gz
## Approx 25% complete for SRR1039520_2.fastq.gz
## Approx 30% complete for SRR1039520_2.fastq.gz
## Approx 35% complete for SRR1039520_2.fastq.gz
## Approx 40% complete for SRR1039520_2.fastq.gz
## Approx 45% complete for SRR1039520_2.fastq.gz
## Approx 50% complete for SRR1039520_2.fastq.gz
## Approx 55% complete for SRR1039520_2.fastq.gz
## Approx 60% complete for SRR1039520_2.fastq.gz
## Approx 65% complete for SRR1039520_2.fastq.gz
## Approx 70% complete for SRR1039520_2.fastq.gz
## Approx 75% complete for SRR1039520_2.fastq.gz
## Approx 80% complete for SRR1039520_2.fastq.gz
## Approx 85% complete for SRR1039520_2.fastq.gz
## Approx 90% complete for SRR1039520_2.fastq.gz
## Approx 95% complete for SRR1039520_2.fastq.gz
## Approx 100% complete for SRR1039520_2.fastq.gz
## Analysis complete for SRR1039520_2.fastq.gz
## Started analysis of SRR1039521_1.fastq.gz
## Approx 5% complete for SRR1039521_1.fastq.gz
## Approx 10% complete for SRR1039521_1.fastq.gz
## Approx 15% complete for SRR1039521_1.fastq.gz
## Approx 20% complete for SRR1039521_1.fastq.gz
## Approx 25% complete for SRR1039521_1.fastq.gz
## Approx 30% complete for SRR1039521_1.fastq.gz
## Approx 35% complete for SRR1039521_1.fastq.gz
## Approx 40% complete for SRR1039521_1.fastq.gz
## Approx 45% complete for SRR1039521_1.fastq.gz
## Approx 50% complete for SRR1039521_1.fastq.gz
## Approx 55% complete for SRR1039521_1.fastq.gz
## Approx 60% complete for SRR1039521_1.fastq.gz
## Approx 65% complete for SRR1039521_1.fastq.gz
## Approx 70% complete for SRR1039521_1.fastq.gz
## Approx 75% complete for SRR1039521_1.fastq.gz
## Approx 80% complete for SRR1039521_1.fastq.gz
## Approx 85% complete for SRR1039521_1.fastq.gz
## Approx 90% complete for SRR1039521_1.fastq.gz
## Approx 95% complete for SRR1039521_1.fastq.gz
## Approx 100% complete for SRR1039521_1.fastq.gz
## Analysis complete for SRR1039521_1.fastq.gz
## Started analysis of SRR1039521_2.fastq.gz
## Approx 5% complete for SRR1039521_2.fastq.gz
## Approx 10% complete for SRR1039521_2.fastq.gz
## Approx 15% complete for SRR1039521_2.fastq.gz
## Approx 20% complete for SRR1039521_2.fastq.gz
## Approx 25% complete for SRR1039521_2.fastq.gz
## Approx 30% complete for SRR1039521_2.fastq.gz
## Approx 35% complete for SRR1039521_2.fastq.gz
## Approx 40% complete for SRR1039521_2.fastq.gz
## Approx 45% complete for SRR1039521_2.fastq.gz
## Approx 50% complete for SRR1039521_2.fastq.gz
## Approx 55% complete for SRR1039521_2.fastq.gz
## Approx 60% complete for SRR1039521_2.fastq.gz
## Approx 65% complete for SRR1039521_2.fastq.gz
## Approx 70% complete for SRR1039521_2.fastq.gz
## Approx 75% complete for SRR1039521_2.fastq.gz
## Approx 80% complete for SRR1039521_2.fastq.gz
## Approx 85% complete for SRR1039521_2.fastq.gz
## Approx 90% complete for SRR1039521_2.fastq.gz
## Approx 95% complete for SRR1039521_2.fastq.gz
## Approx 100% complete for SRR1039521_2.fastq.gz
## Analysis complete for SRR1039521_2.fastq.gz</code></pre>
<p>See what was produced by FastQC.</p>
<div class="sourceCode" id="cb47"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb47-1"><a href="mini-project.html#cb47-1" tabindex="-1"></a><span class="fu">ls</span> fastqc/</span></code></pre></div>
<pre><code>## SRR1039520_1_fastqc.html
## SRR1039520_1_fastqc.zip
## SRR1039520_2_fastqc.html
## SRR1039520_2_fastqc.zip
## SRR1039521_1_fastqc.html
## SRR1039521_1_fastqc.zip
## SRR1039521_2_fastqc.html
## SRR1039521_2_fastqc.zip</code></pre>
</div>
<div id="set-up-a-job-script-to-run-salmon-to-align-the-reads" class="section level2 hasAnchor" number="3.7">
<h2><span class="header-section-number">3.7</span> <strong>Set up a job script to run Salmon to align the reads</strong><a href="mini-project.html#set-up-a-job-script-to-run-salmon-to-align-the-reads" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>First, we <code>exit</code> from our interactive job because we want to get back on to the submit node to submit a non-interactive job to run Salmon.</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb49-1"><a href="mini-project.html#cb49-1" tabindex="-1"></a><span class="co"># You should see one job when you run this command, corresponding to your interactive job.</span></span>
<span id="cb49-2"><a href="mini-project.html#cb49-2" tabindex="-1"></a><span class="ex">squeue</span> <span class="at">--me</span></span>
<span id="cb49-3"><a href="mini-project.html#cb49-3" tabindex="-1"></a></span>
<span id="cb49-4"><a href="mini-project.html#cb49-4" tabindex="-1"></a><span class="co"># Exit the interactive job</span></span>
<span id="cb49-5"><a href="mini-project.html#cb49-5" tabindex="-1"></a><span class="bu">exit</span></span>
<span id="cb49-6"><a href="mini-project.html#cb49-6" tabindex="-1"></a></span>
<span id="cb49-7"><a href="mini-project.html#cb49-7" tabindex="-1"></a><span class="co"># Now you should see no jobs when you run this command because the interactive job has ended.</span></span>
<span id="cb49-8"><a href="mini-project.html#cb49-8" tabindex="-1"></a><span class="ex">squeue</span> <span class="at">--me</span></span>
<span id="cb49-9"><a href="mini-project.html#cb49-9" tabindex="-1"></a></span>
<span id="cb49-10"><a href="mini-project.html#cb49-10" tabindex="-1"></a><span class="co"># Go back to your project directory</span></span>
<span id="cb49-11"><a href="mini-project.html#cb49-11" tabindex="-1"></a><span class="bu">cd</span> /varidata/researchtemp/hpctmp/BBC_workshop_Oct2024_II/<span class="op">&lt;</span>username<span class="op">&gt;</span></span></code></pre></div>
<p>Below is a SLURM job script to run Salmon. For now, do not worry about how the code works. Copy the code and paste it into a new file. Save it as <code>run_salmon.sh</code> in your username directory. If you have issues with this task, you can download the job script directly using the command, <code>wget https://raw.githubusercontent.com/vari-bbc/Intro_to_Linux_for_HPC/refs/heads/main/resources/sec2/run_salmon.sh</code>.</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb50-1"><a href="mini-project.html#cb50-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb50-2"><a href="mini-project.html#cb50-2" tabindex="-1"></a></span>
<span id="cb50-3"><a href="mini-project.html#cb50-3" tabindex="-1"></a><span class="co">#SBATCH --export=NONE</span></span>
<span id="cb50-4"><a href="mini-project.html#cb50-4" tabindex="-1"></a><span class="co">#SBATCH -J run_salmon</span></span>
<span id="cb50-5"><a href="mini-project.html#cb50-5" tabindex="-1"></a><span class="co">#SBATCH -o run_salmon.o</span></span>
<span id="cb50-6"><a href="mini-project.html#cb50-6" tabindex="-1"></a><span class="co">#SBATCH -e run_salmon.e</span></span>
<span id="cb50-7"><a href="mini-project.html#cb50-7" tabindex="-1"></a><span class="co">#SBATCH --ntasks 4</span></span>
<span id="cb50-8"><a href="mini-project.html#cb50-8" tabindex="-1"></a><span class="co">#SBATCH --time 1:00:00</span></span>
<span id="cb50-9"><a href="mini-project.html#cb50-9" tabindex="-1"></a><span class="co">#SBATCH --mem=31G</span></span>
<span id="cb50-10"><a href="mini-project.html#cb50-10" tabindex="-1"></a></span>
<span id="cb50-11"><a href="mini-project.html#cb50-11" tabindex="-1"></a><span class="va">start_time</span><span class="op">=</span><span class="va">$(</span><span class="fu">date</span> +<span class="st">&quot;%T&quot;</span><span class="va">)</span></span>
<span id="cb50-12"><a href="mini-project.html#cb50-12" tabindex="-1"></a></span>
<span id="cb50-13"><a href="mini-project.html#cb50-13" tabindex="-1"></a><span class="co"># You need to navigate to your project directory. Conveniently, the $SLURM_SUBMIT_DIR variable stores the path for where the job was submitted.</span></span>
<span id="cb50-14"><a href="mini-project.html#cb50-14" tabindex="-1"></a><span class="bu">cd</span> <span class="va">${SLURM_SUBMIT_DIR}</span></span>
<span id="cb50-15"><a href="mini-project.html#cb50-15" tabindex="-1"></a></span>
<span id="cb50-16"><a href="mini-project.html#cb50-16" tabindex="-1"></a><span class="ex">module</span> load bbc2/salmon/salmon-1.10.0</span>
<span id="cb50-17"><a href="mini-project.html#cb50-17" tabindex="-1"></a></span>
<span id="cb50-18"><a href="mini-project.html#cb50-18" tabindex="-1"></a><span class="co"># Typically, you would have to first build an index before doing the aligning, but we have done this for you already. Here, we store the path to the index file in a variable called &#39;salmon_idx&#39;.</span></span>
<span id="cb50-19"><a href="mini-project.html#cb50-19" tabindex="-1"></a><span class="va">salmon_idx</span><span class="op">=</span><span class="st">&quot;/varidata/research/projects/bbc/versioned_references/2022-03-08_14.47.50_v9/data/hg38_gencode/indexes/salmon/hg38_gencode/&quot;</span></span>
<span id="cb50-20"><a href="mini-project.html#cb50-20" tabindex="-1"></a></span>
<span id="cb50-21"><a href="mini-project.html#cb50-21" tabindex="-1"></a><span class="co"># make output directory for salmon</span></span>
<span id="cb50-22"><a href="mini-project.html#cb50-22" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> salmon</span>
<span id="cb50-23"><a href="mini-project.html#cb50-23" tabindex="-1"></a></span>
<span id="cb50-24"><a href="mini-project.html#cb50-24" tabindex="-1"></a><span class="co"># This is called a for loop. We use this to run salmon quant on all the samples, one at a time. It is more efficient to run salmon on each sample &quot;in parallel&quot; but we will not do that today.</span></span>
<span id="cb50-25"><a href="mini-project.html#cb50-25" tabindex="-1"></a><span class="cf">for</span> samp_id <span class="kw">in</span> SRR1039520 SRR1039521</span>
<span id="cb50-26"><a href="mini-project.html#cb50-26" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb50-27"><a href="mini-project.html#cb50-27" tabindex="-1"></a>    <span class="ex">salmon</span> quant <span class="at">-p</span> <span class="va">${SLURM_NTASKS}</span> <span class="at">-l</span> A <span class="at">-i</span> <span class="va">$salmon_idx</span> <span class="at">-1</span> fastqs/<span class="va">${samp_id}</span>_1.fastq.gz <span class="at">-2</span> fastqs/<span class="va">${samp_id}</span>_2.fastq.gz <span class="at">-o</span> salmon/<span class="va">${samp_id}</span> <span class="at">--validateMappings</span></span>
<span id="cb50-28"><a href="mini-project.html#cb50-28" tabindex="-1"></a></span>
<span id="cb50-29"><a href="mini-project.html#cb50-29" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb50-30"><a href="mini-project.html#cb50-30" tabindex="-1"></a></span>
<span id="cb50-31"><a href="mini-project.html#cb50-31" tabindex="-1"></a><span class="va">end_time</span><span class="op">=</span><span class="va">$(</span><span class="fu">date</span> +<span class="st">&quot;%T&quot;</span><span class="va">)</span></span>
<span id="cb50-32"><a href="mini-project.html#cb50-32" tabindex="-1"></a><span class="bu">echo</span> <span class="st">&quot;Start time: </span><span class="va">$start_time</span><span class="st">&quot;</span></span>
<span id="cb50-33"><a href="mini-project.html#cb50-33" tabindex="-1"></a><span class="bu">echo</span> <span class="st">&quot;End time: </span><span class="va">$end_time</span><span class="st">&quot;</span></span></code></pre></div>
</div>
<div id="submit-a-job" class="section level2 hasAnchor" number="3.8">
<h2><span class="header-section-number">3.8</span> <strong>Submit a Job</strong><a href="mini-project.html#submit-a-job" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Type <code>ls</code> to ensure <code>run_salmon.sh</code> file exist in the username directory.</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb51-1"><a href="mini-project.html#cb51-1" tabindex="-1"></a><span class="fu">ls</span></span></code></pre></div>
<p>Use the <code>sbatch</code> command to submit the job.</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb52-1"><a href="mini-project.html#cb52-1" tabindex="-1"></a><span class="ex">sbatch</span> <span class="at">-p</span> short run_salmon.sh</span></code></pre></div>
<p>Users can check if the job is running with the following command. The job should take about two minutes to complete.</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb53-1"><a href="mini-project.html#cb53-1" tabindex="-1"></a><span class="ex">squeue</span> <span class="at">--me</span></span></code></pre></div>
</div>
<div id="examine-the-standard-output-stdout-and-standard-error-stderr-log-files" class="section level2 hasAnchor" number="3.9">
<h2><span class="header-section-number">3.9</span> <strong>Examine the Standard Output (stdout) and Standard Error (stderr) log files</strong><a href="mini-project.html#examine-the-standard-output-stdout-and-standard-error-stderr-log-files" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>It is good practice to check the job logs after your job is done to ensure that the job completed successfully. If there is an error, the output files may not be reliable or could be incomplete.</p>
<div class="sourceCode" id="cb54"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb54-1"><a href="mini-project.html#cb54-1" tabindex="-1"></a><span class="fu">tail</span> run_salmon.e</span></code></pre></div>
<pre><code>## [2025-10-03 10:29:55.282] [jointLog] [info] Marked 0 weighted equivalence classes as degenerate
## [2025-10-03 10:29:55.309] [jointLog] [info] iteration = 0 | max rel diff. = 203.028
## [2025-10-03 10:29:57.974] [jointLog] [info] iteration = 100 | max rel diff. = 19.6442
## [2025-10-03 10:30:00.719] [jointLog] [info] iteration = 200 | max rel diff. = 0.462189
## [2025-10-03 10:30:03.319] [jointLog] [info] iteration = 300 | max rel diff. = 1.92755
## [2025-10-03 10:30:05.909] [jointLog] [info] iteration = 400 | max rel diff. = 0.0347574
## [2025-10-03 10:30:07.538] [jointLog] [info] iteration = 463 | max rel diff. = 0.00959217
## [2025-10-03 10:30:07.581] [jointLog] [info] Finished optimizer
## [2025-10-03 10:30:07.581] [jointLog] [info] writing output</code></pre>
<div class="sourceCode" id="cb56"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb56-1"><a href="mini-project.html#cb56-1" tabindex="-1"></a><span class="fu">tail</span> run_salmon.o</span></code></pre></div>
<pre><code>## Start time: 10:28:40
## End time: 10:30:09</code></pre>
</div>
<div id="use-grep-to-find-the-tpms-for-a-specific-gene" class="section level2 hasAnchor" number="3.10">
<h2><span class="header-section-number">3.10</span> <strong>Use grep to find the TPMs for a specific gene</strong><a href="mini-project.html#use-grep-to-find-the-tpms-for-a-specific-gene" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>As an example, let’s try to extract out the TPMs (Transcripts per Million) for MUC1. These values can be found in the <code>quant.sf</code> file in each sample’s folder.</p>
<p>First, let’s take a look at one of these files to figure out the format of these files.</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb58-1"><a href="mini-project.html#cb58-1" tabindex="-1"></a><span class="fu">head</span> salmon/SRR1039520/quant.sf</span></code></pre></div>
<pre><code>## Name Length  EffectiveLength TPM NumReads
## ENST00000456328.2    1657    1501.734    0.000000    0.000
## ENST00000450305.2    632 476.867 0.000000    0.000
## ENST00000488147.1    1351    1195.734    0.000000    0.000
## ENST00000619216.1    68  2.021   0.000000    0.000
## ENST00000473358.1    712 556.827 0.000000    0.000
## ENST00000469289.1    535 379.986 0.000000    0.000
## ENST00000607096.1    138 24.859  0.000000    0.000
## ENST00000417324.1    1187    1031.734    0.000000    0.000
## ENST00000461467.1    590 434.911 0.000000    0.000</code></pre>
<p>From the output above, we can see that the TPMs are in the 4th column of this file. The canonical transcript for <a href="http://useast.ensembl.org/Homo_sapiens/Gene/Summary?db=core;g=ENSG00000185499;r=1:155185824-155192916">MUC1</a> is ENST00000620103, so we will search for that using <code>grep</code>.</p>
<div class="sourceCode" id="cb60"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb60-1"><a href="mini-project.html#cb60-1" tabindex="-1"></a><span class="fu">grep</span> <span class="st">&#39;ENST00000620103&#39;</span> salmon/SRR1039520/quant.sf</span></code></pre></div>
<pre><code>## ENST00000620103.4    1811    1655.734    0.000000    0.000</code></pre>
<p>Look for MUC1 across all the samples at the same time. We can see that ‘SRR1039521’ has a TPM of 13.1 for MUC1 compared to 0 for ‘SRR1039520’. Recall that the fastq files for this exercise were subsetted to a very small number of reads so don’t interpret these results seriously.</p>
<div class="sourceCode" id="cb62"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb62-1"><a href="mini-project.html#cb62-1" tabindex="-1"></a><span class="fu">grep</span> <span class="st">&#39;ENST00000620103&#39;</span> salmon/<span class="pp">*</span>/quant.sf</span></code></pre></div>
<pre><code>## salmon/SRR1039520/quant.sf:ENST00000620103.4 1811    1655.734    0.000000    0.000
## salmon/SRR1039521/quant.sf:ENST00000620103.4 1811    1655.512    13.182825   5.741</code></pre>
</div>
<div id="bonus-use-an-interactive-job-to-run-multiqc-on-the-salmon-and-fastqc-output" class="section level2 hasAnchor" number="3.11">
<h2><span class="header-section-number">3.11</span> <strong>BONUS: Use an interactive job to run multiQC on the Salmon and FastQC output</strong><a href="mini-project.html#bonus-use-an-interactive-job-to-run-multiqc-on-the-salmon-and-fastqc-output" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In this final step, users will start an interactive job to perform multiQC to collect and summarize the outcomes from FastQC and Salmon, then evaluate the quality of the fastq sequences with the reference transcriptome(mapping rate).</p>
<p>Start an interactive job.</p>
<div class="sourceCode" id="cb64"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb64-1"><a href="mini-project.html#cb64-1" tabindex="-1"></a><span class="ex">srun</span> <span class="at">--nodes</span><span class="op">=</span>1 <span class="at">--ntasks-per-node</span><span class="op">=</span>1 <span class="at">--time</span><span class="op">=</span>00:30:00 <span class="at">--pty</span> bash</span></code></pre></div>
<p>Navigate to the project directory.</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb65-1"><a href="mini-project.html#cb65-1" tabindex="-1"></a><span class="bu">cd</span> /varidata/researchtemp/hpctmp/BBC_workshop_Oct2024_II/<span class="op">&lt;</span>username<span class="op">&gt;</span></span></code></pre></div>
<p>Load the environment module for multiQC. Run multiQC, which will summarize the results from FastQC and Salmon and output the results into a new directory called <code>multiqc/</code>.</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb66-1"><a href="mini-project.html#cb66-1" tabindex="-1"></a><span class="ex">module</span> load bbc2/multiqc/multiqc-1.14</span>
<span id="cb66-2"><a href="mini-project.html#cb66-2" tabindex="-1"></a></span>
<span id="cb66-3"><a href="mini-project.html#cb66-3" tabindex="-1"></a><span class="ex">multiqc</span> <span class="at">--outdir</span> multiqc .</span></code></pre></div>
<pre><code>## 
##   /// MultiQC 🔍 | v1.14
## 
## |           multiqc | MultiQC Version v1.31 now available!
## |           multiqc | Search path : /varidata/research/projects/bbc/research/Intro_to_Linux_for_HPC_devel
## |         searching | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 143/143  
## |            fastqc | Found 4 reports
## |           multiqc | Compressing plot data
## |           multiqc | Report      : multiqc/multiqc_report.html
## |           multiqc | Data        : multiqc/multiqc_data
## |           multiqc | MultiQC complete</code></pre>
<p>List the contents of the <code>multiqc</code> directory.</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb68-1"><a href="mini-project.html#cb68-1" tabindex="-1"></a><span class="fu">ls</span> multiqc</span></code></pre></div>
<pre><code>## multiqc_data
## multiqc_report.html</code></pre>
<p>Note the newly created <code>multiqc_report.html</code> file. Try to view this file in your preferred internet browser. If you have mounted the HPC file system to your computer, you can simply double-click on this file. Alternatively, you can copy this file to your computer’s local storage first and then open it.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="basic-command-line-utilities-on-the-hpc.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="assets/gitbook-2.6.7/js/app.min.js"></script>
<script src="assets/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="assets/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="assets/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="assets/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="assets/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="assets/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="assets/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": true,
    "facebook": false,
    "twitter": false,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": false
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": null,
    "text": null
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": null,
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  },
  "toolbar": {
    "position": "static"
  }
});
});
</script>

</body>

</html>
